// amexio-ng-extensions/dashboard/dashboard.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(require('@angular/core'),require('@angular/common'),exports, require('@angular/core'), require('@angular/common'), require('rxjs/Observable')) :
	typeof define === 'function' && define.amd ? define(['@angular/core','@angular/common','exports', '@angular/core', '@angular/common', 'rxjs/Observable'], factory) :
	(factory(global.ng.core,global.ng.common,(global.dashboard = {}),global._angular_core,global._angular_common,global.rxjs_Observable));
}(this, (function (ɵngcc0,ɵngcc1,exports,_angular_core,_angular_common,rxjs_Observable) { 
function GaugeChartComponent_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 2);
} }
var _c0 = ["*"];
function DataPointsComponent_tr_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "tr");
    ɵngcc0.ɵɵelementStart(1, "td");
    ɵngcc0.ɵɵprojection(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵattribute("colspan", ctx_r0.colspan);
} }
function DataPointsComponent_td_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "td");
    ɵngcc0.ɵɵprojection(1, 1);
    ɵngcc0.ɵɵelementEnd();
} }
function DataPointsComponent_td_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "td");
    ɵngcc0.ɵɵprojection(1, 2);
    ɵngcc0.ɵɵelementEnd();
} }
function DataPointsComponent_td_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "td");
    ɵngcc0.ɵɵprojection(1, 3);
    ɵngcc0.ɵɵelementEnd();
} }
function DataPointsComponent_tr_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "tr");
    ɵngcc0.ɵɵelementStart(1, "td");
    ɵngcc0.ɵɵprojection(2, 4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r4 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵattribute("colspan", ctx_r4.colspan);
} }
var _c1 = [[["amexio-north"]], [["amexio-west"]], [["amexio-center"]], [["amexio-east"]], [["amexio-south"]]];
var _c2 = ["amexio-north", "amexio-west", "amexio-center", "amexio-east", "amexio-south"];
'use strict';

/*
 * Copyright 2016-2017 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Author - Sagar Jadhav
 *
 */
var DashBoardTitle = (function () {
    function DashBoardTitle() {
    }
    /**
     * @return {?}
     */
    DashBoardTitle.prototype.ngOnInit = function () {
    };
DashBoardTitle.ɵfac = function DashBoardTitle_Factory(t) { return new (t || DashBoardTitle)(); };
DashBoardTitle.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DashBoardTitle, selectors: [["amexio-dashboard-title"]], inputs: { title: "title", titlePosition: "titlePosition", titleColor: "titleColor", titleFontName: "titleFontName", titleFontSize: "titleFontSize", isTitleBold: "isTitleBold", isTitleItalic: "isTitleItalic" }, decls: 0, vars: 0, template: function DashBoardTitle_Template(rf, ctx) { }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DashBoardTitle, [{
        type: _angular_core.Component,
        args: [{
                selector: 'amexio-dashboard-title',
                template: " "
            }]
    }], function () { return []; }, { title: [{
            type: _angular_core.Input
        }], titlePosition: [{
            type: _angular_core.Input
        }], titleColor: [{
            type: _angular_core.Input
        }], titleFontName: [{
            type: _angular_core.Input
        }], titleFontSize: [{
            type: _angular_core.Input
        }], isTitleBold: [{
            type: _angular_core.Input
        }], isTitleItalic: [{
            type: _angular_core.Input
        }] }); })();
    return DashBoardTitle;
}());
/**
 * @nocollapse
 */
DashBoardTitle.ctorParameters = function () { return []; };
DashBoardTitle.propDecorators = {
    'title': [{ type: _angular_core.Input },],
    'titlePosition': [{ type: _angular_core.Input },],
    'titleColor': [{ type: _angular_core.Input },],
    'titleFontName': [{ type: _angular_core.Input },],
    'titleFontSize': [{ type: _angular_core.Input },],
    'isTitleBold': [{ type: _angular_core.Input },],
    'isTitleItalic': [{ type: _angular_core.Input },],
};

/**
 * Created by pratik on 17/8/17.
 */
var DashboardLoaderService = (function () {
    function DashboardLoaderService() {
        this.chartPackage = {
            AreaChart: 'corechart',
            Bar: 'bar',
            BarChart: 'corechart',
            ColumnChart: 'corechart',
            PieChart: 'corechart',
            Gantt: 'gantt',
            Gauge: 'gauge',
            GeoChart: 'geochart',
            Histogram: 'corechart',
            Line: 'line',
            LineChart: 'corechart',
            Map: 'map',
            Timeline: 'timeline',
            BubbleChart: 'corechart',
            CandlestickChart: 'corechart',
            ComboChart: 'corechart',
            TreeMap: 'treemap'
        };
        this.googleScriptLoadingNotifier = new _angular_core.EventEmitter();
        this.isScriptLoading = false;
    }
    /**
     * @param {?} chartName
     * @return {?}
     */
    DashboardLoaderService.prototype.loadCharts = function (chartName) {
        var _this = this;
        return new rxjs_Observable.Observable(function (observer) {
            _this.loadScript().subscribe(function (val) { return console.log(); }, function (error) { return console.error(error); }, function () {
                _this.loadRequiredChart(observer, chartName);
            });
        });
    };
    /**
     * @return {?}
     */
    DashboardLoaderService.prototype.loadScript = function () {
        var _this = this;
        return new rxjs_Observable.Observable(function (observer) {
            if (!_this.isScriptLoading) {
                //check if previously its loaded
                if (typeof google !== 'undefined' && google.charts) {
                    //check if chart package has been loaded using chartPackagename ?
                    _this.loadBaseChart(observer);
                }
                else {
                    _this.isScriptLoading = true;
                    var /** @type {?} */ script = document.createElement('script');
                    script.type = 'text/javascript';
                    script.src = 'https://www.gstatic.com/charts/loader.js';
                    script.async = true;
                    script.defer = true;
                    script.onload = function () {
                        _this.isScriptLoading = false;
                        _this.googleScriptLoadingNotifier.emit(true);
                        _this.loadBaseChart(observer);
                    };
                    script.onerror = function () {
                        observer.error('Could Not Load google Script');
                    };
                    document.getElementsByTagName('head')[0].appendChild(script);
                }
            }
            else {
                _this.googleScriptLoadingNotifier.subscribe(function (loaded) {
                    if (loaded) {
                        _this.loadBaseChart(observer);
                    }
                });
            }
        });
    };
    /**
     * Load Base Chart
     * @param {?} observer
     * @return {?}
     */
    DashboardLoaderService.prototype.loadBaseChart = function (observer) {
        google.charts.load('current', { 'packages': ['corechart'] });
        google.charts.setOnLoadCallback(function () {
            observer.complete();
        });
    };
    /**
     * Load the required charts
     * @param {?} observer
     * @param {?} chartName
     * @return {?}
     */
    DashboardLoaderService.prototype.loadRequiredChart = function (observer, chartName) {
        if (google.visualization.hasOwnProperty(chartName)) {
            observer.complete();
        }
        else {
            google.charts.load('current', { 'packages': [this.chartPackage[chartName]] });
            google.charts.setOnLoadCallback(function () {
                observer.complete();
            });
        }
    };
DashboardLoaderService.ɵfac = function DashboardLoaderService_Factory(t) { return new (t || DashboardLoaderService)(); };
DashboardLoaderService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: DashboardLoaderService, factory: function (t) { return DashboardLoaderService.ɵfac(t); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DashboardLoaderService, [{
        type: _angular_core.Injectable
    }], function () { return []; }, null); })();
    return DashboardLoaderService;
}());
/**
 * @nocollapse
 */
DashboardLoaderService.ctorParameters = function () { return []; };

/*
 * Copyright 2016-2017 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Author - Sagar Jadhav
 *
 */
var GaugeChartComponent = (function () {
    /**
     * @param {?} loader
     */
    function GaugeChartComponent(loader) {
        this.loader = loader;
        this.id = 'amexio-chart-gauge' + Math.floor(Math.random() * 90000) + 10000;
        this.width = '100%';
    }
    /**
     * @return {?}
     */
    GaugeChartComponent.prototype.drawChart = function () {
        this.gaugeData = google.visualization.arrayToDataTable(this.data);
        this.options = {
            width: this.width,
            height: this.height,
            redFrom: this.redColorFrom,
            redTo: this.redColorTo,
            yellowFrom: this.yellowColorFrom,
            yellowTo: this.yellowColorTo,
            minorTicks: this.minorTicks
        };
        this.chart = new google.visualization.Gauge(document.getElementById(this.id));
        this.hasLoaded = true;
        this.chart.draw(this.gaugeData, this.options);
    };
    /**
     * @return {?}
     */
    GaugeChartComponent.prototype.ngAfterContentInit = function () {
        this.chartTitleArray = this.chartTitleComp.toArray();
        //take first component
        if (this.chartTitleArray.length == 1) {
            this.chartTitleComponent = this.chartTitleArray.pop();
        }
    };
    /**
     * @return {?}
     */
    GaugeChartComponent.prototype.ngOnInit = function () {
        var _this = this;
        this.hasLoaded = false;
        this.loader.loadCharts('Gauge').subscribe(function (value) { return console.log(); }, function (errror) { return console.error(errror); }, function () {
            _this.drawChart();
        });
    };
    /**
     * @param {?} event
     * @return {?}
     */
    GaugeChartComponent.prototype.onResize = function (event) {
        this.drawChart();
    };
GaugeChartComponent.ɵfac = function GaugeChartComponent_Factory(t) { return new (t || GaugeChartComponent)(ɵngcc0.ɵɵdirectiveInject(DashboardLoaderService)); };
GaugeChartComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: GaugeChartComponent, selectors: [["amexio-dashboard-gauge"]], contentQueries: function GaugeChartComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, DashBoardTitle, false);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.chartTitleComp = _t);
    } }, inputs: { width: "width", height: "height", data: "data", redColorFrom: "redColorFrom", redColorTo: "redColorTo", yellowColorFrom: "yellowColorFrom", yellowColorTo: "yellowColorTo", minorTicks: "minorTicks" }, decls: 2, vars: 6, consts: [[3, "resize"], ["class", "lmask", 4, "ngIf"], [1, "lmask"]], template: function GaugeChartComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵlistener("resize", function GaugeChartComponent_Template_div_resize_0_listener($event) { return ctx.onResize($event); }, false, ɵngcc0.ɵɵresolveWindow);
        ɵngcc0.ɵɵtemplate(1, GaugeChartComponent_div_1_Template, 1, 0, "div", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("width", ctx.width)("height", ctx.height);
        ɵngcc0.ɵɵattribute("id", ctx.id);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.hasLoaded);
    } }, directives: [ɵngcc1.NgIf], styles: [".lmask[_ngcontent-%COMP%] {\n    position: absolute;\n    height: 100%;\n    width: 100%;\n    background-color: #000;\n    bottom: 0;\n    left: 0;\n    right: 0;\n    top: 0;\n    z-index: 9999;\n    opacity: 0.4;\n  }\n  .lmask.fixed[_ngcontent-%COMP%] {\n    position: fixed;\n  }\n  .lmask[_ngcontent-%COMP%]:before {\n    content: '';\n    background-color: transparent;\n    border: 5px solid rgba(0, 183, 229, 0.9);\n    opacity: .9;\n    border-right: 5px solid transparent;\n    border-left: 5px solid transparent;\n    border-radius: 50px;\n    box-shadow: 0 0 35px #2187e7;\n    width: 50px;\n    height: 50px;\n    -moz-animation: spinPulse 1s infinite ease-in-out;\n    -webkit-animation: spinPulse 1s infinite linear;\n    margin: -25px 0 0 -25px;\n    position: absolute;\n    top: 50%;\n    left: 50%;\n  }\n  .lmask[_ngcontent-%COMP%]:after {\n    content: '';\n    background-color: transparent;\n    border: 5px solid rgba(0, 183, 229, 0.9);\n    opacity: .9;\n    border-left: 5px solid transparent;\n    border-right: 5px solid transparent;\n    border-radius: 50px;\n    box-shadow: 0 0 15px #2187e7;\n    width: 30px;\n    height: 30px;\n    -moz-animation: spinoffPulse 1s infinite linear;\n    -webkit-animation: spinoffPulse 1s infinite linear;\n    margin: -15px 0 0 -15px;\n    position: absolute;\n    top: 50%;\n    left: 50%;\n  }\n\n  @-moz-keyframes spinPulse {\n    0% {\n      -moz-transform: rotate(160deg);\n      opacity: 0;\n      box-shadow: 0 0 1px #2187e7;\n    }\n    50% {\n      -moz-transform: rotate(145deg);\n      opacity: 1;\n    }\n    100% {\n      -moz-transform: rotate(-320deg);\n      opacity: 0;\n    }\n  }\n  @-moz-keyframes spinoffPulse {\n    0% {\n      -moz-transform: rotate(0deg);\n    }\n    100% {\n      -moz-transform: rotate(360deg);\n    }\n  }\n  @-webkit-keyframes spinPulse {\n    0% {\n      -webkit-transform: rotate(160deg);\n      opacity: 0;\n      box-shadow: 0 0 1px #2187e7;\n    }\n    50% {\n      -webkit-transform: rotate(145deg);\n      opacity: 1;\n    }\n    100% {\n      -webkit-transform: rotate(-320deg);\n      opacity: 0;\n    }\n  }\n  @-webkit-keyframes spinoffPulse {\n    0% {\n      -webkit-transform: rotate(0deg);\n    }\n    100% {\n      -webkit-transform: rotate(360deg);\n    }\n  }"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(GaugeChartComponent, [{
        type: _angular_core.Component,
        args: [{
                selector: 'amexio-dashboard-gauge',
                template: "\n      <div [attr.id]=\"id\"\n           [style.width]=\"width\"\n           [style.height]=\"height\" (window:resize)=\"onResize($event)\" >\n        <div *ngIf=\"!hasLoaded\" class=\"lmask\">\n        </div>\n      </div>\n  ",
                styles: [".lmask {\n    position: absolute;\n    height: 100%;\n    width: 100%;\n    background-color: #000;\n    bottom: 0;\n    left: 0;\n    right: 0;\n    top: 0;\n    z-index: 9999;\n    opacity: 0.4;\n  }\n  .lmask.fixed {\n    position: fixed;\n  }\n  .lmask:before {\n    content: '';\n    background-color: transparent;\n    border: 5px solid rgba(0, 183, 229, 0.9);\n    opacity: .9;\n    border-right: 5px solid transparent;\n    border-left: 5px solid transparent;\n    border-radius: 50px;\n    box-shadow: 0 0 35px #2187e7;\n    width: 50px;\n    height: 50px;\n    -moz-animation: spinPulse 1s infinite ease-in-out;\n    -webkit-animation: spinPulse 1s infinite linear;\n    margin: -25px 0 0 -25px;\n    position: absolute;\n    top: 50%;\n    left: 50%;\n  }\n  .lmask:after {\n    content: '';\n    background-color: transparent;\n    border: 5px solid rgba(0, 183, 229, 0.9);\n    opacity: .9;\n    border-left: 5px solid transparent;\n    border-right: 5px solid transparent;\n    border-radius: 50px;\n    box-shadow: 0 0 15px #2187e7;\n    width: 30px;\n    height: 30px;\n    -moz-animation: spinoffPulse 1s infinite linear;\n    -webkit-animation: spinoffPulse 1s infinite linear;\n    margin: -15px 0 0 -15px;\n    position: absolute;\n    top: 50%;\n    left: 50%;\n  }\n\n  @-moz-keyframes spinPulse {\n    0% {\n      -moz-transform: rotate(160deg);\n      opacity: 0;\n      box-shadow: 0 0 1px #2187e7;\n    }\n    50% {\n      -moz-transform: rotate(145deg);\n      opacity: 1;\n    }\n    100% {\n      -moz-transform: rotate(-320deg);\n      opacity: 0;\n    }\n  }\n  @-moz-keyframes spinoffPulse {\n    0% {\n      -moz-transform: rotate(0deg);\n    }\n    100% {\n      -moz-transform: rotate(360deg);\n    }\n  }\n  @-webkit-keyframes spinPulse {\n    0% {\n      -webkit-transform: rotate(160deg);\n      opacity: 0;\n      box-shadow: 0 0 1px #2187e7;\n    }\n    50% {\n      -webkit-transform: rotate(145deg);\n      opacity: 1;\n    }\n    100% {\n      -webkit-transform: rotate(-320deg);\n      opacity: 0;\n    }\n  }\n  @-webkit-keyframes spinoffPulse {\n    0% {\n      -webkit-transform: rotate(0deg);\n    }\n    100% {\n      -webkit-transform: rotate(360deg);\n    }\n  }\n\n  "]
            }]
    }], function () { return [{ type: DashboardLoaderService }]; }, { width: [{
            type: _angular_core.Input
        }], height: [{
            type: _angular_core.Input
        }], data: [{
            type: _angular_core.Input
        }], redColorFrom: [{
            type: _angular_core.Input
        }], redColorTo: [{
            type: _angular_core.Input
        }], yellowColorFrom: [{
            type: _angular_core.Input
        }], yellowColorTo: [{
            type: _angular_core.Input
        }], minorTicks: [{
            type: _angular_core.Input
        }], chartTitleComp: [{
            type: _angular_core.ContentChildren,
            args: [DashBoardTitle]
        }] }); })();
    return GaugeChartComponent;
}());
/**
 * @nocollapse
 */
GaugeChartComponent.ctorParameters = function () { return [
    { type: DashboardLoaderService, },
]; };
GaugeChartComponent.propDecorators = {
    'width': [{ type: _angular_core.Input },],
    'height': [{ type: _angular_core.Input },],
    'data': [{ type: _angular_core.Input },],
    'redColorFrom': [{ type: _angular_core.Input },],
    'redColorTo': [{ type: _angular_core.Input },],
    'yellowColorFrom': [{ type: _angular_core.Input },],
    'yellowColorTo': [{ type: _angular_core.Input },],
    'minorTicks': [{ type: _angular_core.Input },],
    'chartTitleComp': [{ type: _angular_core.ContentChildren, args: [DashBoardTitle,] },],
};

/**
 * Created by ketangote on 7/25/17.
 */
var DataPointCenterComponent = (function () {
    function DataPointCenterComponent() {
    }
    /**
     * @return {?}
     */
    DataPointCenterComponent.prototype.ngOnInit = function () {
    };
DataPointCenterComponent.ɵfac = function DataPointCenterComponent_Factory(t) { return new (t || DataPointCenterComponent)(); };
DataPointCenterComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DataPointCenterComponent, selectors: [["amexio-center"]], inputs: { contentalign: "contentalign", backgroundColor: "backgroundColor", fontColor: "fontColor", width: "width", height: "height", cClass: "cClass" }, ngContentSelectors: _c0, decls: 2, vars: 10, consts: [[3, "ngClass"]], template: function DataPointCenterComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("background-color", ctx.backgroundColor)("color", ctx.fontColor)("width", ctx.width)("height", ctx.height);
        ɵngcc0.ɵɵproperty("ngClass", ctx.cClass);
        ɵngcc0.ɵɵattribute("align", ctx.contentalign);
    } }, directives: [ɵngcc1.NgClass], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DataPointCenterComponent, [{
        type: _angular_core.Component,
        args: [{
                selector: 'amexio-center',
                template: "\n\n    <div [ngClass]=\"cClass\" [attr.align]=\"contentalign\" [style.background-color]=\"backgroundColor\" [style.color]=\"fontColor\" [style.width]=\"width\" [style.height]=\"height\">\n      <ng-content></ng-content>\n    </div>\n\n\n  "
            }]
    }], function () { return []; }, { contentalign: [{
            type: _angular_core.Input
        }], backgroundColor: [{
            type: _angular_core.Input
        }], fontColor: [{
            type: _angular_core.Input
        }], width: [{
            type: _angular_core.Input
        }], height: [{
            type: _angular_core.Input
        }], cClass: [{
            type: _angular_core.Input
        }] }); })();
    return DataPointCenterComponent;
}());
/**
 * @nocollapse
 */
DataPointCenterComponent.ctorParameters = function () { return []; };
DataPointCenterComponent.propDecorators = {
    'contentalign': [{ type: _angular_core.Input },],
    'backgroundColor': [{ type: _angular_core.Input },],
    'fontColor': [{ type: _angular_core.Input },],
    'width': [{ type: _angular_core.Input },],
    'height': [{ type: _angular_core.Input },],
    'cClass': [{ type: _angular_core.Input },],
};

/**
 * Created by ketangote on 8/16/17.
 */
var DataPointsComponent = (function () {
    function DataPointsComponent() {
        this.colspan = 1;
    }
    /**
     * @return {?}
     */
    DataPointsComponent.prototype.ngOnInit = function () {
        if (this.west)
            this.colspan++;
        if (this.east)
            this.colspan++;
    };
DataPointsComponent.ɵfac = function DataPointsComponent_Factory(t) { return new (t || DataPointsComponent)(); };
DataPointsComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DataPointsComponent, selectors: [["amexio-datapoints"]], inputs: { north: "north", south: "south", west: "west", center: "center", east: "east", backgroundColor: "backgroundColor", fontColor: "fontColor" }, ngContentSelectors: _c2, decls: 8, vars: 9, consts: [[2, "padding", "5px"], ["width", "100%"], [4, "ngIf"]], template: function DataPointsComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef(_c1);
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "table", 1);
        ɵngcc0.ɵɵtemplate(2, DataPointsComponent_tr_2_Template, 3, 1, "tr", 2);
        ɵngcc0.ɵɵelementStart(3, "tr");
        ɵngcc0.ɵɵtemplate(4, DataPointsComponent_td_4_Template, 2, 0, "td", 2);
        ɵngcc0.ɵɵtemplate(5, DataPointsComponent_td_5_Template, 2, 0, "td", 2);
        ɵngcc0.ɵɵtemplate(6, DataPointsComponent_td_6_Template, 2, 0, "td", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(7, DataPointsComponent_tr_7_Template, 3, 1, "tr", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.backgroundColor)("color", ctx.fontColor);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.north);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.west);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.center);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.east);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.south);
    } }, directives: [ɵngcc1.NgIf], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DataPointsComponent, [{
        type: _angular_core.Component,
        args: [{
                selector: 'amexio-datapoints',
                template: "      \n      \n  <div style=\"padding: 5px;\">\n    <table width=\"100%\"  [style.background-color]=\"backgroundColor\" [style.color]=\"fontColor\">\n      <tr *ngIf=\"north\">\n        <td [attr.colspan]=\"colspan\">\n          <ng-content select=\"amexio-north\"></ng-content>\n        </td>\n      </tr>\n      <tr>\n        <td *ngIf=\"west\">\n          <ng-content select=\"amexio-west\"></ng-content>\n        </td>\n        <td *ngIf=\"center\">\n          <ng-content select=\"amexio-center\"></ng-content>\n        </td>\n        <td *ngIf=\"east\">\n          <ng-content select=\"amexio-east\"></ng-content>\n        </td>\n      </tr>\n      <tr  *ngIf=\"south\">\n        <td [attr.colspan]=\"colspan\">\n          <ng-content select=\"amexio-south\"></ng-content>\n        </td>\n      </tr>\n    </table>\n  </div>\n\n\n  "
            }]
    }], function () { return []; }, { north: [{
            type: _angular_core.Input
        }], south: [{
            type: _angular_core.Input
        }], west: [{
            type: _angular_core.Input
        }], center: [{
            type: _angular_core.Input
        }], east: [{
            type: _angular_core.Input
        }], backgroundColor: [{
            type: _angular_core.Input
        }], fontColor: [{
            type: _angular_core.Input
        }] }); })();
    return DataPointsComponent;
}());
/**
 * @nocollapse
 */
DataPointsComponent.ctorParameters = function () { return []; };
DataPointsComponent.propDecorators = {
    'north': [{ type: _angular_core.Input },],
    'south': [{ type: _angular_core.Input },],
    'west': [{ type: _angular_core.Input },],
    'center': [{ type: _angular_core.Input },],
    'east': [{ type: _angular_core.Input },],
    'backgroundColor': [{ type: _angular_core.Input },],
    'fontColor': [{ type: _angular_core.Input },],
};

/**
 * Created by ketangote on 7/25/17.
 */
var DataPointEastComponent = (function () {
    function DataPointEastComponent() {
    }
    /**
     * @return {?}
     */
    DataPointEastComponent.prototype.ngOnInit = function () {
    };
DataPointEastComponent.ɵfac = function DataPointEastComponent_Factory(t) { return new (t || DataPointEastComponent)(); };
DataPointEastComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DataPointEastComponent, selectors: [["amexio-east"]], inputs: { contentalign: "contentalign", backgroundColor: "backgroundColor", fontColor: "fontColor", width: "width", height: "height", cClass: "cClass" }, ngContentSelectors: _c0, decls: 2, vars: 10, consts: [[3, "ngClass"]], template: function DataPointEastComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("background-color", ctx.backgroundColor)("color", ctx.fontColor)("width", ctx.width)("height", ctx.height);
        ɵngcc0.ɵɵproperty("ngClass", ctx.cClass);
        ɵngcc0.ɵɵattribute("align", ctx.contentalign);
    } }, directives: [ɵngcc1.NgClass], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DataPointEastComponent, [{
        type: _angular_core.Component,
        args: [{
                selector: 'amexio-east',
                template: "\n\n    <div [ngClass]=\"cClass\" [attr.align]=\"contentalign\" [style.background-color]=\"backgroundColor\" [style.color]=\"fontColor\" [style.width]=\"width\" [style.height]=\"height\">\n      <ng-content></ng-content>\n    </div>\n\n\n  "
            }]
    }], function () { return []; }, { contentalign: [{
            type: _angular_core.Input
        }], backgroundColor: [{
            type: _angular_core.Input
        }], fontColor: [{
            type: _angular_core.Input
        }], width: [{
            type: _angular_core.Input
        }], height: [{
            type: _angular_core.Input
        }], cClass: [{
            type: _angular_core.Input
        }] }); })();
    return DataPointEastComponent;
}());
/**
 * @nocollapse
 */
DataPointEastComponent.ctorParameters = function () { return []; };
DataPointEastComponent.propDecorators = {
    'contentalign': [{ type: _angular_core.Input },],
    'backgroundColor': [{ type: _angular_core.Input },],
    'fontColor': [{ type: _angular_core.Input },],
    'width': [{ type: _angular_core.Input },],
    'height': [{ type: _angular_core.Input },],
    'cClass': [{ type: _angular_core.Input },],
};

/**
 * Created by ketangote on 7/25/17.
 */
var DataPointNorthComponent = (function () {
    function DataPointNorthComponent() {
    }
    /**
     * @return {?}
     */
    DataPointNorthComponent.prototype.ngOnInit = function () {
    };
DataPointNorthComponent.ɵfac = function DataPointNorthComponent_Factory(t) { return new (t || DataPointNorthComponent)(); };
DataPointNorthComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DataPointNorthComponent, selectors: [["amexio-north"]], inputs: { contentalign: "contentalign", backgroundColor: "backgroundColor", fontColor: "fontColor", width: "width", height: "height", cClass: "cClass" }, ngContentSelectors: _c0, decls: 2, vars: 10, consts: [[3, "ngClass"]], template: function DataPointNorthComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("background-color", ctx.backgroundColor)("color", ctx.fontColor)("width", ctx.width)("height", ctx.height);
        ɵngcc0.ɵɵproperty("ngClass", ctx.cClass);
        ɵngcc0.ɵɵattribute("align", ctx.contentalign);
    } }, directives: [ɵngcc1.NgClass], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DataPointNorthComponent, [{
        type: _angular_core.Component,
        args: [{
                selector: 'amexio-north',
                template: "\n\n    <div [ngClass]=\"cClass\" [attr.align]=\"contentalign\" [style.background-color]=\"backgroundColor\" [style.color]=\"fontColor\" [style.width]=\"width\" [style.height]=\"height\">\n      <ng-content></ng-content>\n    </div>\n    \n\n\n  "
            }]
    }], function () { return []; }, { contentalign: [{
            type: _angular_core.Input
        }], backgroundColor: [{
            type: _angular_core.Input
        }], fontColor: [{
            type: _angular_core.Input
        }], width: [{
            type: _angular_core.Input
        }], height: [{
            type: _angular_core.Input
        }], cClass: [{
            type: _angular_core.Input
        }] }); })();
    return DataPointNorthComponent;
}());
/**
 * @nocollapse
 */
DataPointNorthComponent.ctorParameters = function () { return []; };
DataPointNorthComponent.propDecorators = {
    'contentalign': [{ type: _angular_core.Input },],
    'backgroundColor': [{ type: _angular_core.Input },],
    'fontColor': [{ type: _angular_core.Input },],
    'width': [{ type: _angular_core.Input },],
    'height': [{ type: _angular_core.Input },],
    'cClass': [{ type: _angular_core.Input },],
};

/**
 * Created by ketangote on 7/25/17.
 */
var DataPointSouthComponent = (function () {
    function DataPointSouthComponent() {
    }
    /**
     * @return {?}
     */
    DataPointSouthComponent.prototype.ngOnInit = function () {
    };
DataPointSouthComponent.ɵfac = function DataPointSouthComponent_Factory(t) { return new (t || DataPointSouthComponent)(); };
DataPointSouthComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DataPointSouthComponent, selectors: [["amexio-south"]], inputs: { contentalign: "contentalign", backgroundColor: "backgroundColor", fontColor: "fontColor", width: "width", height: "height", cClass: "cClass" }, ngContentSelectors: _c0, decls: 2, vars: 10, consts: [[3, "ngClass"]], template: function DataPointSouthComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("background-color", ctx.backgroundColor)("color", ctx.fontColor)("width", ctx.width)("height", ctx.height);
        ɵngcc0.ɵɵproperty("ngClass", ctx.cClass);
        ɵngcc0.ɵɵattribute("align", ctx.contentalign);
    } }, directives: [ɵngcc1.NgClass], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DataPointSouthComponent, [{
        type: _angular_core.Component,
        args: [{
                selector: 'amexio-south',
                template: "\n\n    <div [ngClass]=\"cClass\" [attr.align]=\"contentalign\" [style.background-color]=\"backgroundColor\" [style.color]=\"fontColor\" [style.width]=\"width\" [style.height]=\"height\">\n      <ng-content></ng-content>\n    </div>\n\n\n  "
            }]
    }], function () { return []; }, { contentalign: [{
            type: _angular_core.Input
        }], backgroundColor: [{
            type: _angular_core.Input
        }], fontColor: [{
            type: _angular_core.Input
        }], width: [{
            type: _angular_core.Input
        }], height: [{
            type: _angular_core.Input
        }], cClass: [{
            type: _angular_core.Input
        }] }); })();
    return DataPointSouthComponent;
}());
/**
 * @nocollapse
 */
DataPointSouthComponent.ctorParameters = function () { return []; };
DataPointSouthComponent.propDecorators = {
    'contentalign': [{ type: _angular_core.Input },],
    'backgroundColor': [{ type: _angular_core.Input },],
    'fontColor': [{ type: _angular_core.Input },],
    'width': [{ type: _angular_core.Input },],
    'height': [{ type: _angular_core.Input },],
    'cClass': [{ type: _angular_core.Input },],
};

/**
 * Created by ketangote on 7/25/17.
 */
var DataPointWestComponent = (function () {
    function DataPointWestComponent() {
    }
    /**
     * @return {?}
     */
    DataPointWestComponent.prototype.ngOnInit = function () {
    };
DataPointWestComponent.ɵfac = function DataPointWestComponent_Factory(t) { return new (t || DataPointWestComponent)(); };
DataPointWestComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DataPointWestComponent, selectors: [["amexio-west"]], inputs: { contentalign: "contentalign", backgroundColor: "backgroundColor", fontColor: "fontColor", width: "width", height: "height", cClass: "cClass" }, ngContentSelectors: _c0, decls: 2, vars: 10, consts: [[3, "ngClass"]], template: function DataPointWestComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("background-color", ctx.backgroundColor)("color", ctx.fontColor)("width", ctx.width)("height", ctx.height);
        ɵngcc0.ɵɵproperty("ngClass", ctx.cClass);
        ɵngcc0.ɵɵattribute("align", ctx.contentalign);
    } }, directives: [ɵngcc1.NgClass], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DataPointWestComponent, [{
        type: _angular_core.Component,
        args: [{
                selector: 'amexio-west',
                template: "\n\n    <div [ngClass]=\"cClass\" [attr.align]=\"contentalign\" [style.background-color]=\"backgroundColor\" [style.color]=\"fontColor\" [style.width]=\"width\" [style.height]=\"height\">\n      <ng-content></ng-content>\n    </div>\n\n\n  "
            }]
    }], function () { return []; }, { contentalign: [{
            type: _angular_core.Input
        }], backgroundColor: [{
            type: _angular_core.Input
        }], fontColor: [{
            type: _angular_core.Input
        }], width: [{
            type: _angular_core.Input
        }], height: [{
            type: _angular_core.Input
        }], cClass: [{
            type: _angular_core.Input
        }] }); })();
    return DataPointWestComponent;
}());
/**
 * @nocollapse
 */
DataPointWestComponent.ctorParameters = function () { return []; };
DataPointWestComponent.propDecorators = {
    'contentalign': [{ type: _angular_core.Input },],
    'backgroundColor': [{ type: _angular_core.Input },],
    'fontColor': [{ type: _angular_core.Input },],
    'width': [{ type: _angular_core.Input },],
    'height': [{ type: _angular_core.Input },],
    'cClass': [{ type: _angular_core.Input },],
};

var AmexioDashboardModule = (function () {
    function AmexioDashboardModule() {
    }
    /**
     * @return {?}
     */
    AmexioDashboardModule.forRoot = function () {
        return {
            ngModule: AmexioDashboardModule,
            providers: [DashboardLoaderService]
        };
    };
AmexioDashboardModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AmexioDashboardModule });
AmexioDashboardModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function AmexioDashboardModule_Factory(t) { return new (t || AmexioDashboardModule)(); }, providers: [DashboardLoaderService], imports: [[
            _angular_common.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AmexioDashboardModule, { declarations: [DashBoardTitle, GaugeChartComponent, DataPointCenterComponent, DataPointsComponent, DataPointEastComponent, DataPointNorthComponent, DataPointSouthComponent, DataPointWestComponent], imports: [ɵngcc1.CommonModule], exports: [DashBoardTitle, GaugeChartComponent, DataPointCenterComponent, DataPointsComponent, DataPointEastComponent, DataPointNorthComponent, DataPointSouthComponent, DataPointWestComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AmexioDashboardModule, [{
        type: _angular_core.NgModule,
        args: [{
                imports: [
                    _angular_common.CommonModule
                ],
                declarations: [
                    DashBoardTitle,
                    GaugeChartComponent,
                    DataPointCenterComponent,
                    DataPointsComponent,
                    DataPointEastComponent,
                    DataPointNorthComponent,
                    DataPointSouthComponent,
                    DataPointWestComponent
                ],
                exports: [
                    DashBoardTitle,
                    GaugeChartComponent,
                    DataPointCenterComponent,
                    DataPointsComponent,
                    DataPointEastComponent,
                    DataPointNorthComponent,
                    DataPointSouthComponent,
                    DataPointWestComponent
                ],
                providers: [DashboardLoaderService]
            }]
    }], function () { return []; }, null); })();
    return AmexioDashboardModule;
}());
/**
 * @nocollapse
 */
AmexioDashboardModule.ctorParameters = function () { return []; };

exports.AmexioDashboardModule = AmexioDashboardModule;
exports.DashBoardTitle = DashBoardTitle;
exports.GaugeChartComponent = GaugeChartComponent;
exports.DataPointCenterComponent = DataPointCenterComponent;
exports.DataPointsComponent = DataPointsComponent;
exports.DataPointEastComponent = DataPointEastComponent;
exports.DataPointNorthComponent = DataPointNorthComponent;
exports.DataPointSouthComponent = DataPointSouthComponent;
exports.DataPointWestComponent = DataPointWestComponent;
exports.DashboardLoaderService = DashboardLoaderService;

Object.defineProperty(exports, '__esModule', { value: true });

})));
