// amexio-ng-extensions/enterprise/media/media.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(require('@angular/core'),require('@angular/common'),require('@angular/forms'),require('@angular/http'),exports, require('@angular/core'), require('@angular/common'), require('@angular/http'), require('@angular/forms')) :
	typeof define === 'function' && define.amd ? define(['@angular/core','@angular/common','@angular/forms','@angular/http','exports', '@angular/core', '@angular/common', '@angular/http', '@angular/forms'], factory) :
	(factory(global.ng.core,global.ng.common,global.ng.forms,global.ng.http,(global.media = {}),global._angular_core,global._angular_common,global._angular_http,global._angular_forms));
}(this, (function (ɵngcc0,ɵngcc1,ɵngcc2,ɵngcc3,exports,_angular_core,_angular_common,_angular_http,_angular_forms) { 
function ContentComponent_span_1_Template(rf, ctx) { if (rf & 1) {
    var _r8 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "span", 12);
    ɵngcc0.ɵɵelementStart(1, "i", 13);
    ɵngcc0.ɵɵlistener("click", function ContentComponent_span_1_Template_i_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r8); var ctx_r7 = ɵngcc0.ɵɵnextContext(); return ctx_r7.closeDetailPage(); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} }
function ContentComponent_span_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 14);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1("", ctx_r1.matchPercentage, " Match");
} }
function ContentComponent_span_9_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 15);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r2.ageLimit);
} }
function ContentComponent_ng_container_14_li_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "li");
    ɵngcc0.ɵɵelementStart(1, "strong", 17);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "strong");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var data_r10 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1("", data_r10.key, ": ");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", data_r10.value, "");
} }
function ContentComponent_ng_container_14_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "span");
    ɵngcc0.ɵɵtemplate(2, ContentComponent_ng_container_14_li_2_Template, 5, 2, "li", 16);
    ɵngcc0.ɵɵelement(3, "br");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    var ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r3.contents);
} }
function ContentComponent_li_16_Template(rf, ctx) { if (rf & 1) {
    var _r12 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li");
    ɵngcc0.ɵɵelementStart(1, "amexio-ee-btn", 18);
    ɵngcc0.ɵɵlistener("onClick", function ContentComponent_li_16_Template_amexio_ee_btn_onClick_1_listener() { ɵngcc0.ɵɵrestoreView(_r12); var ctx_r11 = ɵngcc0.ɵɵnextContext(); return ctx_r11.playVideo(); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r4 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("cClass", "buttoncustom")("label", "Play")("type", "danger")("icon", "fa fa-play")("tooltipMessage", "play")("size", ctx_r4.size);
} }
function ContentComponent_li_17_Template(rf, ctx) { if (rf & 1) {
    var _r14 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li");
    ɵngcc0.ɵɵelementStart(1, "amexio-ee-btn", 19);
    ɵngcc0.ɵɵlistener("onClick", function ContentComponent_li_17_Template_amexio_ee_btn_onClick_1_listener() { ɵngcc0.ɵɵrestoreView(_r14); var ctx_r13 = ɵngcc0.ɵɵnextContext(); return ctx_r13.addToList(); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r5 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("cClass", "secondarybutton")("label", "MY LIST")("icon", "fa fa-plus")("type", "secondary")("tooltipMessage", "My List")("size", ctx_r5.size);
} }
function ContentComponent_ng_container_24_Template(rf, ctx) { if (rf & 1) {
    var _r16 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "amexio-ee-rating-input", 20);
    ɵngcc0.ɵɵlistener("ngModelChange", function ContentComponent_ng_container_24_Template_amexio_ee_rating_input_ngModelChange_1_listener($event) { ɵngcc0.ɵɵrestoreView(_r16); var ctx_r15 = ɵngcc0.ɵɵnextContext(); return ctx_r15.rate = $event; });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    var ctx_r6 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngModel", ctx_r6.rate)("max", ctx_r6.max)("readonly", ctx_r6.isReadonly);
} }
var _c0 = function (a0) { return { "background-image": a0 }; };
function ButtonMediaComponent_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelement(1, "i", 2);
    ɵngcc0.ɵɵtext(2, "\u00A0 ");
    ɵngcc0.ɵɵelementContainerEnd();
} }
function ButtonMediaComponent_ng_container_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelement(1, "i", 3);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMap(ctx_r1.iconStyleClass);
} }
function RatingMediaComponent_span_1_Template(rf, ctx) { if (rf & 1) {
    var _r4 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵelementStart(1, "i", 2);
    ɵngcc0.ɵɵlistener("mouseenter", function RatingMediaComponent_span_1_Template_i_mouseenter_1_listener() { ɵngcc0.ɵɵrestoreView(_r4); var item_r1 = ctx.$implicit; var ctx_r3 = ɵngcc0.ɵɵnextContext(); return ctx_r3.setHovered(item_r1); })("mousemove", function RatingMediaComponent_span_1_Template_i_mousemove_1_listener($event) { ɵngcc0.ɵɵrestoreView(_r4); var ctx_r5 = ɵngcc0.ɵɵnextContext(); return ctx_r5.changeHovered($event); })("click", function RatingMediaComponent_span_1_Template_i_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r4); var item_r1 = ctx.$implicit; var ctx_r6 = ɵngcc0.ɵɵnextContext(); return ctx_r6.rate(item_r1); });
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var item_r1 = ctx.$implicit;
    var index_r2 = ctx.index;
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMapInterpolate2("", ctx_r0.iconClass, " half", ctx_r0.calculateWidth(item_r1), "");
    ɵngcc0.ɵɵproperty("title", ctx_r0.titles[index_r2] || item_r1);
    ɵngcc0.ɵɵattribute("data-icon", ctx_r0.fullIcon);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r0.emptyIcon);
} }
function MultiMediaCarousel_ng_container_8_div_1_span_10_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 26);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var item_r3 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(item_r3.title);
} }
function MultiMediaCarousel_ng_container_8_div_1_div_21_Template(rf, ctx) { if (rf & 1) {
    var _r10 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 27);
    ɵngcc0.ɵɵelementStart(1, "div", 28);
    ɵngcc0.ɵɵlistener("click", function MultiMediaCarousel_ng_container_8_div_1_div_21_Template_div_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r10); var item_r3 = ɵngcc0.ɵɵnextContext().$implicit; var ctx_r8 = ɵngcc0.ɵɵnextContext(2); return ctx_r8.openDetailsSection(item_r3); });
    ɵngcc0.ɵɵelement(2, "i", 29);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} }
var _c1 = function (a0) { return { "visibility": a0 }; };
var _c2 = function (a0, a1) { return { "tile__details": a0, "tile__details_v": a1 }; };
function MultiMediaCarousel_ng_container_8_div_1_Template(rf, ctx) { if (rf & 1) {
    var _r12 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 12);
    ɵngcc0.ɵɵelementStart(1, "div", 13);
    ɵngcc0.ɵɵlistener("click", function MultiMediaCarousel_ng_container_8_div_1_Template_div_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r12); var item_r3 = ctx.$implicit; var ctx_r11 = ɵngcc0.ɵɵnextContext(2); return ctx_r11.loadVideo(item_r3); });
    ɵngcc0.ɵɵelement(2, "img", 14);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "div");
    ɵngcc0.ɵɵelement(4, "i", 15);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(5, "div", 16);
    ɵngcc0.ɵɵelementStart(6, "div", 17);
    ɵngcc0.ɵɵelementStart(7, "div", 18);
    ɵngcc0.ɵɵelementStart(8, "div", 19);
    ɵngcc0.ɵɵlistener("click", function MultiMediaCarousel_ng_container_8_div_1_Template_div_click_8_listener() { ɵngcc0.ɵɵrestoreView(_r12); var item_r3 = ctx.$implicit; var ctx_r13 = ɵngcc0.ɵɵnextContext(2); return ctx_r13.loadVideo(item_r3); });
    ɵngcc0.ɵɵelement(9, "i", 20);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(10, MultiMediaCarousel_ng_container_8_div_1_span_10_Template, 2, 1, "span", 21);
    ɵngcc0.ɵɵelement(11, "br");
    ɵngcc0.ɵɵelementStart(12, "span", 22);
    ɵngcc0.ɵɵtext(13);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(14);
    ɵngcc0.ɵɵelementStart(15, "span", 23);
    ɵngcc0.ɵɵtext(16);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(17);
    ɵngcc0.ɵɵelement(18, "br");
    ɵngcc0.ɵɵelementStart(19, "div", 24);
    ɵngcc0.ɵɵtext(20);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(21, MultiMediaCarousel_ng_container_8_div_1_div_21_Template, 3, 0, "div", 25);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var item_r3 = ctx.$implicit;
    var ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵattribute("src", item_r3.img, ɵngcc0.ɵɵsanitizeUrl);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMap(item_r3.selected);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(13, _c1, item_r3.selected ? "visible" : "hidden"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction2(15, _c2, ctx_r2.carouselStyle == "horizontal", ctx_r2.carouselStyle == "vertical"));
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵproperty("ngIf", item_r3.title);
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate1("", item_r3.matchPercentage, " Match");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", item_r3.releaseYear, " ");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(item_r3.ageLimit);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", item_r3.seasonNo, "");
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate(item_r3.desc);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", item_r3.content);
} }
function MultiMediaCarousel_ng_container_8_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtemplate(1, MultiMediaCarousel_ng_container_8_div_1_Template, 22, 18, "div", 11);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r0.data);
} }
function MultiMediaCarousel_div_12_Template(rf, ctx) { if (rf & 1) {
    var _r15 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 30);
    ɵngcc0.ɵɵelementStart(1, "amexio-ee-content", 31);
    ɵngcc0.ɵɵlistener("onWatchClick", function MultiMediaCarousel_div_12_Template_amexio_ee_content_onWatchClick_1_listener() { ɵngcc0.ɵɵrestoreView(_r15); var ctx_r14 = ɵngcc0.ɵɵnextContext(); return ctx_r14.playVideo(ctx_r14.videoUrl); })("onCloseClick", function MultiMediaCarousel_div_12_Template_amexio_ee_content_onCloseClick_1_listener() { ɵngcc0.ɵɵrestoreView(_r15); var ctx_r16 = ɵngcc0.ɵɵnextContext(); return ctx_r16.closeDetailPage(); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵattribute("id", ctx_r1.elementId + "carousel-detail-content");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("bgImgUrl", ctx_r1.currentDetailsImagePath)("title", ctx_r1.currentDetailsTitle)("description", ctx_r1.currentDetailsDesc)("contents", ctx_r1.currentContent)("videoLink", ctx_r1.videoUrl)("rate", ctx_r1.rate)("enableWatch", true)("enableMyList", true)("closeEnable", true)("seasonNo", ctx_r1.seasonNo)("releaseYear", ctx_r1.releaseYear)("ageLimit", ctx_r1.ageLimit)("matchPercentage", ctx_r1.matchPercentage);
} }
var _c3 = function (a0, a1) { return { "controls": a0, "controls-v": a1 }; };
'use strict';

/**
 * Created by dattaram on 23/8/17.
 */
var ContentComponent = (function () {
    function ContentComponent() {
        this.enableWatch = false;
        this.enableMyList = false;
        this.onWatchClick = new _angular_core.EventEmitter();
        this.onAddListClick = new _angular_core.EventEmitter();
        this.onLikeClick = new _angular_core.EventEmitter();
        this.onUnlikeLikeClick = new _angular_core.EventEmitter();
        this.onCloseClick = new _angular_core.EventEmitter();
        this.closeEnable = false;
        this.smallScreen = false;
        console.log($(window).width());
    }
    /**
     * @return {?}
     */
    ContentComponent.prototype.ngOnInit = function () {
        if (window.innerWidth < 768) {
            this.smallScreen = true;
            this.size = 'small';
        }
        else {
            this.smallScreen = false;
            this.size = 'default';
        }
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    ContentComponent.prototype.ngOnChanges = function (changes) {
        if (changes.videoLink && !changes.videoLink.isFirstChange()) {
            this.videoLink = changes.videoLink.currentValue;
            console.log(this.videoLink);
        }
    };
    /**
     * @return {?}
     */
    ContentComponent.prototype.playVideo = function () {
        this.onWatchClick.emit(this.videoLink);
    };
    /**
     * @return {?}
     */
    ContentComponent.prototype.addToList = function () {
        this.overviewData = {
            'title': this.title,
            'description': this.description,
            'Video Link': this.videoLink,
            'rate': this.rate
        };
        this.onAddListClick.emit(this.overviewData);
    };
    /**
     * @param {?} event
     * @return {?}
     */
    ContentComponent.prototype.onResize = function (event) {
        if (event.target.innerWidth < 995) {
            this.size = 'small';
            this.smallScreen = true;
        }
        else {
            this.size = 'default';
            this.smallScreen = false;
        }
    };
    /**
     * @return {?}
     */
    ContentComponent.prototype.getClassName = function () {
        if (this.smallScreen) {
            return 'buttonSmall';
        }
        else {
            return 'button';
        }
    };
    /**
     * @return {?}
     */
    ContentComponent.prototype.unlikeClick = function () {
        this.overviewData = {
            'title': this.title,
            'rate': this.rate
        };
        this.onUnlikeLikeClick.emit(this.overviewData);
    };
    /**
     * @return {?}
     */
    ContentComponent.prototype.likeClick = function () {
        this.overviewData = {
            'title': this.title,
            'rate': this.rate
        };
        this.onLikeClick.emit(this.overviewData);
    };
    /**
     * @return {?}
     */
    ContentComponent.prototype.closeDetailPage = function () {
        this.onCloseClick.emit(this.title);
    };
ContentComponent.ɵfac = function ContentComponent_Factory(t) { return new (t || ContentComponent)(); };
ContentComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ContentComponent, selectors: [["amexio-ee-content"]], inputs: { enableWatch: "enableWatch", enableMyList: "enableMyList", closeEnable: "closeEnable", videoLink: "videoLink", bgImgUrl: "bgImgUrl", title: "title", description: "description", contents: "contents", rate: "rate", max: "max", isReadonly: "isReadonly", ageLimit: "ageLimit", releaseYear: "releaseYear", seasonNo: "seasonNo", matchPercentage: "matchPercentage" }, outputs: { onWatchClick: "onWatchClick", onAddListClick: "onAddListClick", onLikeClick: "onLikeClick", onUnlikeLikeClick: "onUnlikeLikeClick", onCloseClick: "onCloseClick" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 25, vars: 16, consts: [[1, "maincontent", 3, "ngStyle", "resize"], ["class", "close-button", 4, "ngIf"], [1, "col-lg-6", "col-sm-12", "content-area", 2, "color", "white", "padding-left", "30px"], [1, "title"], [1, "subtitle"], ["class", "match", 4, "ngIf"], ["class", "age", 4, "ngIf"], [4, "ngIf"], [2, "padding-left", "0"], [3, "ngClass", "click"], ["aria-hidden", "true", 1, "fa", "fa-thumbs-o-up", "fa-lg"], ["aria-hidden", "true", 1, "fa", "fa-thumbs-o-down", "fa-lg"], [1, "close-button"], ["aria-hidden", "true", 1, "fa", "fa-times", "fa-lg", 3, "click"], [1, "match"], [1, "age"], [4, "ngFor", "ngForOf"], [2, "color", "#999"], [3, "cClass", "label", "type", "icon", "tooltipMessage", "size", "onClick"], [3, "cClass", "label", "icon", "type", "tooltipMessage", "size", "onClick"], ["name", "rate", 3, "ngModel", "max", "readonly", "ngModelChange"]], template: function ContentComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵlistener("resize", function ContentComponent_Template_div_resize_0_listener($event) { return ctx.onResize($event); }, false, ɵngcc0.ɵɵresolveWindow);
        ɵngcc0.ɵɵtemplate(1, ContentComponent_span_1_Template, 2, 0, "span", 1);
        ɵngcc0.ɵɵelementStart(2, "div", 2);
        ɵngcc0.ɵɵelementStart(3, "span", 3);
        ɵngcc0.ɵɵtext(4);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelement(5, "br");
        ɵngcc0.ɵɵelementStart(6, "span", 4);
        ɵngcc0.ɵɵtemplate(7, ContentComponent_span_7_Template, 2, 1, "span", 5);
        ɵngcc0.ɵɵtext(8);
        ɵngcc0.ɵɵtemplate(9, ContentComponent_span_9_Template, 2, 1, "span", 6);
        ɵngcc0.ɵɵtext(10);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelement(11, "br");
        ɵngcc0.ɵɵtext(12);
        ɵngcc0.ɵɵelement(13, "br");
        ɵngcc0.ɵɵtemplate(14, ContentComponent_ng_container_14_Template, 4, 1, "ng-container", 7);
        ɵngcc0.ɵɵelementStart(15, "ul", 8);
        ɵngcc0.ɵɵtemplate(16, ContentComponent_li_16_Template, 2, 6, "li", 7);
        ɵngcc0.ɵɵtemplate(17, ContentComponent_li_17_Template, 2, 6, "li", 7);
        ɵngcc0.ɵɵelementStart(18, "li");
        ɵngcc0.ɵɵelementStart(19, "a", 9);
        ɵngcc0.ɵɵlistener("click", function ContentComponent_Template_a_click_19_listener() { return ctx.likeClick(); });
        ɵngcc0.ɵɵelement(20, "i", 10);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(21, "li");
        ɵngcc0.ɵɵelementStart(22, "a", 9);
        ɵngcc0.ɵɵlistener("click", function ContentComponent_Template_a_click_22_listener() { return ctx.unlikeClick(); });
        ɵngcc0.ɵɵelement(23, "i", 11);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(24, ContentComponent_ng_container_24_Template, 2, 3, "ng-container", 7);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(14, _c0, "url(" + ctx.bgImgUrl + ")"));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.closeEnable);
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate(ctx.title);
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("ngIf", ctx.matchPercentage);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.releaseYear, " ");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.ageLimit);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.seasonNo, "");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.description, "");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.contents);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.enableWatch);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.enableMyList);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngClass", ctx.getClassName());
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("ngClass", ctx.getClassName());
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.rate && ctx.max);
    } }, directives: function () { return [ɵngcc1.NgStyle, ɵngcc1.NgIf, ɵngcc1.NgClass, ɵngcc1.NgForOf, ButtonMediaComponent, RatingMediaComponent, ɵngcc2.NgControlStatus, ɵngcc2.NgModel]; }, styles: ["ul[_ngcontent-%COMP%]   li[_ngcontent-%COMP%] {\n              display:inline;\n          }\n\n          .maincontent[_ngcontent-%COMP%] {\n              position: relative;\n              width: 100%; \n              height: 65vh;\n              width: 100%;\n              background-size: 100% 100%;\n              background-repeat: no-repeat;\n          }\n          .maincontent[_ngcontent-%COMP%]   div[_ngcontent-%COMP%] {\n              position: absolute;\n              top: 10px;\n              left: 0;\n              width: 100%;\n          }\n\n          .title[_ngcontent-%COMP%]{\n              width: 100%;\n              font-size: 180%;\n              font-weight: 700;\n              line-height: 130%;\n              color: #fff;\n              text-shadow: 2px 4px 3px rgba(0,0,0,0.3);\n          }\n          @media screen and (min-width: 1400px){\n              .maincontent[_ngcontent-%COMP%]   div[_ngcontent-%COMP%] {\n                  height: 32vw;\n                  padding-bottom: 20px;\n              }\n          }\n          @media (max-width: 799px)\n          {\n              .maincontent[_ngcontent-%COMP%]   div[_ngcontent-%COMP%] {\n                  font-size: 11px;\n              }\n\n              .maincontent[_ngcontent-%COMP%]   div[_ngcontent-%COMP%] {\n                  top: 10px;\n                  width: 33vw;\n                  min-width: 330px;\n                  color: #999;\n                  line-height: 1.3;\n              }\n              .maincontent[_ngcontent-%COMP%]   div[_ngcontent-%COMP%]   table[_ngcontent-%COMP%] {\n                  height: 50px;\n              }\n          }\n          @media(max-width: 480px) {\n\n              img[_ngcontent-%COMP%]{\n                  width: 170px;\n                  height: 35px;\n              }\n              .title[_ngcontent-%COMP%]{\n                  width: 100%;\n                  font-size: 180%;\n                  font-weight: 700;\n                  line-height: 130%;\n                  color: #fff;\n                  text-shadow: 2px 4px 3px rgba(0,0,0,0.3);\n              }\n              h3[_ngcontent-%COMP%] {\n                  font-size: 9pt;\n              }\n              span[_ngcontent-%COMP%]{\n                  font-size: 6pt;\n              }\n              li[_ngcontent-%COMP%]   b[_ngcontent-%COMP%]{\n                  font-size: 6pt;\n              }\n              li[_ngcontent-%COMP%] {\n                  font-size: 6pt;\n              }\n              .maincontent[_ngcontent-%COMP%] {\n                  position: relative;\n                  width: 100%; \n                  height: 35vh;\n                  width: 100%;\n                  background-size: 100% 100%;\n                  background-repeat: no-repeat;\n              }\n          }\n          header[_ngcontent-%COMP%]{\n              width: 100%;\n              padding: 0px 48px;\n              background-image: linear-gradient(rgba(0, 0, 0, 0.0), #141414);\n              background-size: cover;\n              background-position: center;\n              height: 100vh;\n          }\n          header[_ngcontent-%COMP%]   .billboard[_ngcontent-%COMP%]{\n              margin-top: 40px;\n              width: 40%;\n              padding: 2px;\n              font-size: 110%;\n              line-height: 160%;\n          }\n          header[_ngcontent-%COMP%]   .billboard[_ngcontent-%COMP%]   .tab-area[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]{\n              width: 120px;\n              float: left;\n              font-size: 90%;\n              color: #ffffff;\n              text-align: center;\n              text-transform: uppercase;\n              text-decoration: none;\n              padding: 8px;\n              margin-right: .75em;\n              margin-bottom: 30px;\n              transition: border 0.5s;\n          }\n          header[_ngcontent-%COMP%]   .billboard[_ngcontent-%COMP%]   .tab-area[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]:hover{\n              border-bottom: solid 2px #eeeeee;\n          }\n          header[_ngcontent-%COMP%]   .billboard[_ngcontent-%COMP%]   .tab-area[_ngcontent-%COMP%]   .active[_ngcontent-%COMP%]{\n              border-bottom: solid 2px #e50914;\n          }\n          \n          header[_ngcontent-%COMP%]   .billboard[_ngcontent-%COMP%]   .content-area[_ngcontent-%COMP%]{\n              color: #999;\n              font-size: 100%;\n          }\n          header[_ngcontent-%COMP%]   .billboard[_ngcontent-%COMP%]   .content-area[_ngcontent-%COMP%]   .subtitle[_ngcontent-%COMP%]{\n              font-size: 120%;\n              line-height: 200%;\n          }\n          header[_ngcontent-%COMP%]   .billboard[_ngcontent-%COMP%]   .content-area[_ngcontent-%COMP%]   .credits[_ngcontent-%COMP%]{\n              font-size: 80%;\n          }\n          header[_ngcontent-%COMP%]   .billboard[_ngcontent-%COMP%]   .content-area[_ngcontent-%COMP%]   .credits[_ngcontent-%COMP%]   strong[_ngcontent-%COMP%]{\n              color: #fff;\n          }\n          \n          header[_ngcontent-%COMP%]   .billboard[_ngcontent-%COMP%]   .button[_ngcontent-%COMP%]{\n              display: inline-block;\n              margin-right: .75em;\n              margin-top: 20px;\n              padding: 5px 20px;\n              font-size: 70%;\n              font-weight: 500;\n              text-decoration: none;\n              text-transform: uppercase;\n              color: #fff;\n          }\n          .button[_ngcontent-%COMP%]   i[_ngcontent-%COMP%]{\n              display: inline-block;\n              margin-top: 20px;\n              width: 40px;\n              height: 40px;\n              padding: 10px 0 0 11px;\n              border: solid 2px rgba(255,255,255,.4);\n              border-radius: 100px;\n              font-size: 18px;\n              font-weight: 500;\n              text-decoration: none;\n              text-transform: uppercase;\n              color: #fff;\n              transition: transform 0.5s;\n          }\n          .button[_ngcontent-%COMP%]   i[_ngcontent-%COMP%]:hover{\n              border: solid 2px #fff;\n              transform: scale(1.1);\n          }\n\n          .buttonSmall[_ngcontent-%COMP%]   i[_ngcontent-%COMP%]{\n              display: inline-block;\n              width: 25px;\n              height: 25px;\n              padding: 7px 0 0 5px;\n              border: solid 2px rgba(255,255,255,.4);\n              border-radius: 100px;\n              font-size: 12px;\n              font-weight: 500;\n              text-decoration: none;\n              text-transform: uppercase;\n              color: #fff;\n              transition: transform 0.5s;\n          }\n          .buttonSmall[_ngcontent-%COMP%]   i[_ngcontent-%COMP%]:hover{\n              border: solid 2px #fff;\n              transform: scale(1.1);\n          }\n\n          .close-button[_ngcontent-%COMP%]{\n              cursor: pointer;\n              padding-left: 98%;color:gray;background: radial-gradient(ellipse at top right,rgba(0,0,0,.4) 0,rgba(0,0,0,0)70%,rgba(0,0,0,0) 100%);\n          }"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ContentComponent, [{
        type: _angular_core.Component,
        args: [{
                selector: 'amexio-ee-content',
                template: "\n      <div class=\"maincontent\"  [ngStyle]=\"{'background-image':'url('+bgImgUrl+')'}\" (window:resize)=\"onResize($event)\">\n       <span *ngIf=\"closeEnable\" class=\"close-button\">\n        <i class=\"fa fa-times fa-lg\" (click)=\"closeDetailPage()\" aria-hidden=\"true\"></i>\n      </span>\n          <div class=\"col-lg-6 col-sm-12 content-area\" style=\"color: white;padding-left: 30px;\">\n              <span class=\"title\">{{title}}</span><br>\n              <span class=\"subtitle\"><span class=\"match\" *ngIf=\"matchPercentage\">{{matchPercentage}} Match</span> {{releaseYear}} <span class=\"age\" *ngIf=\"ageLimit\">{{ageLimit}}</span> {{seasonNo}}</span><br>\n              {{description}}<br>\n\n              <ng-container *ngIf=\"contents\">\n              <span>\n                <li *ngFor=\"let data of contents\"><strong style=\"color: #999;\">{{data.key}}: </strong> <strong>  {{data.value}}</strong></li><br>\n          </span>\n              </ng-container>\n              <ul style=\"padding-left: 0\">\n                  <li *ngIf=\"enableWatch\"><amexio-ee-btn [cClass]=\"'buttoncustom'\" [label]=\"'Play'\" [type]=\"'danger'\" [icon]=\"'fa fa-play'\" [tooltipMessage]=\"'play'\" [size]=\"size\" (onClick)=\"playVideo()\"></amexio-ee-btn></li>\n                  <li *ngIf=\"enableMyList\" ><amexio-ee-btn [cClass]=\"'secondarybutton'\" [label]=\"'MY LIST'\" [icon]=\"'fa fa-plus'\" [type]=\"'secondary'\" [tooltipMessage]=\"'My List'\" [size]=\"size\" (onClick)=\"addToList()\"></amexio-ee-btn></li>\n                  <li><a   [ngClass]=\"getClassName()\" (click)=\"likeClick()\"><i class=\"fa fa-thumbs-o-up fa-lg\" aria-hidden=\"true\"></i></a>\n                  </li>\n                  <li><a   [ngClass]=\"getClassName()\" (click)=\"unlikeClick()\"><i class=\"fa fa-thumbs-o-down fa-lg \" aria-hidden=\"true\"></i></a></li>\n\n              </ul>\n              <ng-container *ngIf=\"rate && max\">\n                  <amexio-ee-rating-input [(ngModel)]=\"rate\"\n                                       [max]=\"max\" name=\"rate\"\n                                       [readonly]=\"isReadonly\">\n                  </amexio-ee-rating-input>\n              </ng-container>\n          </div>\n      </div>\n  ",
                styles: [
                    "\n          ul li {\n              display:inline;\n          }\n\n          .maincontent {\n              position: relative;\n              width: 100%; /* for IE 6 */\n              height: 65vh;\n              width: 100%;\n              background-size: 100% 100%;\n              background-repeat: no-repeat;\n          }\n          .maincontent div {\n              position: absolute;\n              top: 10px;\n              left: 0;\n              width: 100%;\n          }\n\n          .title{\n              width: 100%;\n              font-size: 180%;\n              font-weight: 700;\n              line-height: 130%;\n              color: #fff;\n              text-shadow: 2px 4px 3px rgba(0,0,0,0.3);\n          }\n          @media screen and (min-width: 1400px){\n              .maincontent div {\n                  height: 32vw;\n                  padding-bottom: 20px;\n              }\n          }\n          @media (max-width: 799px)\n          {\n              .maincontent div {\n                  font-size: 11px;\n              }\n\n              .maincontent div {\n                  top: 10px;\n                  width: 33vw;\n                  min-width: 330px;\n                  color: #999;\n                  line-height: 1.3;\n              }\n              .maincontent div table {\n                  height: 50px;\n              }\n          }\n          @media(max-width: 480px) {\n\n              img{\n                  width: 170px;\n                  height: 35px;\n              }\n              .title{\n                  width: 100%;\n                  font-size: 180%;\n                  font-weight: 700;\n                  line-height: 130%;\n                  color: #fff;\n                  text-shadow: 2px 4px 3px rgba(0,0,0,0.3);\n              }\n              h3 {\n                  font-size: 9pt;\n              }\n              span{\n                  font-size: 6pt;\n              }\n              li b{\n                  font-size: 6pt;\n              }\n              li {\n                  font-size: 6pt;\n              }\n              .maincontent {\n                  position: relative;\n                  width: 100%; /* for IE 6 */\n                  height: 35vh;\n                  width: 100%;\n                  background-size: 100% 100%;\n                  background-repeat: no-repeat;\n              }\n          }\n          header{\n              width: 100%;\n              padding: 0px 48px;\n              background-image: linear-gradient(rgba(0, 0, 0, 0.0), #141414);\n              background-size: cover;\n              background-position: center;\n              height: 100vh;\n          }\n          header .billboard{\n              margin-top: 40px;\n              width: 40%;\n              padding: 2px;\n              font-size: 110%;\n              line-height: 160%;\n          }\n          header .billboard .tab-area a{\n              width: 120px;\n              float: left;\n              font-size: 90%;\n              color: #ffffff;\n              text-align: center;\n              text-transform: uppercase;\n              text-decoration: none;\n              padding: 8px;\n              margin-right: .75em;\n              margin-bottom: 30px;\n              transition: border 0.5s;\n          }\n          header .billboard .tab-area a:hover{\n              border-bottom: solid 2px #eeeeee;\n          }\n          header .billboard .tab-area .active{\n              border-bottom: solid 2px #e50914;\n          }\n          /* Text Related CSS */\n          header .billboard .content-area{\n              color: #999;\n              font-size: 100%;\n          }\n          header .billboard .content-area .subtitle{\n              font-size: 120%;\n              line-height: 200%;\n          }\n          header .billboard .content-area .credits{\n              font-size: 80%;\n          }\n          header .billboard .content-area .credits strong{\n              color: #fff;\n          }\n          /* Button Related CSS */\n          header .billboard .button{\n              display: inline-block;\n              margin-right: .75em;\n              margin-top: 20px;\n              padding: 5px 20px;\n              font-size: 70%;\n              font-weight: 500;\n              text-decoration: none;\n              text-transform: uppercase;\n              color: #fff;\n          }\n          .button i{\n              display: inline-block;\n              margin-top: 20px;\n              width: 40px;\n              height: 40px;\n              padding: 10px 0 0 11px;\n              border: solid 2px rgba(255,255,255,.4);\n              border-radius: 100px;\n              font-size: 18px;\n              font-weight: 500;\n              text-decoration: none;\n              text-transform: uppercase;\n              color: #fff;\n              transition: transform 0.5s;\n          }\n          .button i:hover{\n              border: solid 2px #fff;\n              transform: scale(1.1);\n          }\n\n          .buttonSmall i{\n              display: inline-block;\n              width: 25px;\n              height: 25px;\n              padding: 7px 0 0 5px;\n              border: solid 2px rgba(255,255,255,.4);\n              border-radius: 100px;\n              font-size: 12px;\n              font-weight: 500;\n              text-decoration: none;\n              text-transform: uppercase;\n              color: #fff;\n              transition: transform 0.5s;\n          }\n          .buttonSmall i:hover{\n              border: solid 2px #fff;\n              transform: scale(1.1);\n          }\n\n          .close-button{\n              cursor: pointer;\n              padding-left: 98%;color:gray;background: radial-gradient(ellipse at top right,rgba(0,0,0,.4) 0,rgba(0,0,0,0)70%,rgba(0,0,0,0) 100%);\n          }\n\n    "
                ]
            }]
    }], function () { return []; }, { enableWatch: [{
            type: _angular_core.Input
        }], enableMyList: [{
            type: _angular_core.Input
        }], onWatchClick: [{
            type: _angular_core.Output
        }], onAddListClick: [{
            type: _angular_core.Output
        }], onLikeClick: [{
            type: _angular_core.Output
        }], onUnlikeLikeClick: [{
            type: _angular_core.Output
        }], onCloseClick: [{
            type: _angular_core.Output
        }], closeEnable: [{
            type: _angular_core.Input
        }], videoLink: [{
            type: _angular_core.Input
        }], bgImgUrl: [{
            type: _angular_core.Input
        }], title: [{
            type: _angular_core.Input
        }], description: [{
            type: _angular_core.Input
        }], contents: [{
            type: _angular_core.Input
        }], rate: [{
            type: _angular_core.Input
        }], max: [{
            type: _angular_core.Input
        }], isReadonly: [{
            type: _angular_core.Input
        }], ageLimit: [{
            type: _angular_core.Input
        }], releaseYear: [{
            type: _angular_core.Input
        }], seasonNo: [{
            type: _angular_core.Input
        }], matchPercentage: [{
            type: _angular_core.Input
        }] }); })();
    return ContentComponent;
}());
/**
 * @nocollapse
 */
ContentComponent.ctorParameters = function () { return []; };
ContentComponent.propDecorators = {
    'bgImgUrl': [{ type: _angular_core.Input },],
    'title': [{ type: _angular_core.Input },],
    'description': [{ type: _angular_core.Input },],
    'contents': [{ type: _angular_core.Input },],
    'videoLink': [{ type: _angular_core.Input },],
    'rate': [{ type: _angular_core.Input },],
    'max': [{ type: _angular_core.Input },],
    'isReadonly': [{ type: _angular_core.Input },],
    'enableWatch': [{ type: _angular_core.Input },],
    'enableMyList': [{ type: _angular_core.Input },],
    'ageLimit': [{ type: _angular_core.Input },],
    'releaseYear': [{ type: _angular_core.Input },],
    'seasonNo': [{ type: _angular_core.Input },],
    'matchPercentage': [{ type: _angular_core.Input },],
    'closeEnable': [{ type: _angular_core.Input },],
    'onWatchClick': [{ type: _angular_core.Output },],
    'onAddListClick': [{ type: _angular_core.Output },],
    'onLikeClick': [{ type: _angular_core.Output },],
    'onUnlikeLikeClick': [{ type: _angular_core.Output },],
    'onCloseClick': [{ type: _angular_core.Output },],
};

/*
 * Copyright 2016-2017 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Author -  Dattaram Gawas
 *
 */
var ButtonMediaComponent = (function () {
    function ButtonMediaComponent() {
        this.onClick = new _angular_core.EventEmitter();
        this.elementId = 'button-' + Math.floor(Math.random() * 90000) + 10000;
    }
    /**
     * @return {?}
     */
    ButtonMediaComponent.prototype.ngOnInit = function () {
        if (this.cClass) {
            this.btnStyleClass = this.cClass + '  btn ';
        }
        else {
            this.btnStyleClass = 'btn ';
        }
        this.btnSizeStyleClass = '';
        if (this.type.toLocaleLowerCase() === 'warning') {
            this.btnStyleClass = this.btnStyleClass + 'btn-warning';
        }
        else if (this.type.toLocaleLowerCase() === 'primary') {
            this.btnStyleClass = this.btnStyleClass + 'btn-primary';
        }
        else if (this.type.toLocaleLowerCase() === 'success') {
            this.btnStyleClass = this.btnStyleClass + 'btn-success';
        }
        else if (this.type.toLocaleLowerCase() === 'danger') {
            this.btnStyleClass = this.btnStyleClass + 'btn-danger';
        }
        else if (this.type.toLocaleLowerCase() === 'link') {
            this.btnStyleClass = this.btnStyleClass + 'btn-link';
        }
        else {
            this.btnStyleClass = this.btnStyleClass + 'btn-secondary';
        }
        this.iconStyleClass = this.icon;
        if (this.tooltipMessage == null) {
            this.hasToolTip = false;
        }
        if (this.size != null) {
            if (this.size === 'large') {
                this.btnStyleClass = this.btnStyleClass.concat(' btn-lg');
            }
            else if (this.size === 'small') {
                this.btnStyleClass = this.btnStyleClass.concat(' btn-sm');
            }
        }
        if (this.block) {
            this.btnStyleClass = this.btnStyleClass.concat(' btn-block');
        }
        if (this.popoverPlacement == null) {
            this.popoverPlacement = 'bottom';
        }
    };
    /**
     * @param {?} change
     * @return {?}
     */
    ButtonMediaComponent.prototype.ngOnChanges = function (change) {
        /*console.log(change.isLoading);
          if(change.isLoading){
            this.disabled = true;
          }*/ // TODO : Fix
    };
    /**
     * @return {?}
     */
    ButtonMediaComponent.prototype.ngAfterViewInit = function () {
        // $('[data-toggle="popover"]').popover();
    };
    /**
     * @param {?} event
     * @return {?}
     */
    ButtonMediaComponent.prototype.btnClick = function (event) {
        this.onClick.emit(event);
        if (this.onClickRoute != null) {
            // this.router.navigate([this.onClickRoute]);
        }
    };
ButtonMediaComponent.ɵfac = function ButtonMediaComponent_Factory(t) { return new (t || ButtonMediaComponent)(); };
ButtonMediaComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ButtonMediaComponent, selectors: [["amexio-ee-btn"]], inputs: { popoverPlacement: "popoverPlacement", label: "label", icon: "icon", type: "type", onClickRoute: "onClickRoute", tooltipMessage: "tooltipMessage", disabled: "disabled", isLoading: "isLoading", size: "size", block: "block", fieldName: "fieldName", cClass: "cClass" }, outputs: { onClick: "onClick" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 4, vars: 9, consts: [["type", "button", "data-toggle", "tooltip", 3, "click"], [4, "ngIf"], ["aria-hidden", "true", 1, "fa", "fa-refresh", "fa-spin"], ["aria-hidden", "true"]], template: function ButtonMediaComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "button", 0);
        ɵngcc0.ɵɵlistener("click", function ButtonMediaComponent_Template_button_click_0_listener($event) { return ctx.btnClick($event); });
        ɵngcc0.ɵɵtemplate(1, ButtonMediaComponent_ng_container_1_Template, 3, 0, "ng-container", 1);
        ɵngcc0.ɵɵtemplate(2, ButtonMediaComponent_ng_container_2_Template, 2, 2, "ng-container", 1);
        ɵngcc0.ɵɵtext(3);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.btnStyleClass);
        ɵngcc0.ɵɵattribute("fieldName", ctx.fieldName)("disabled", ctx.disabled ? true : null)("data-placement", ctx.popoverPlacement)("title", ctx.tooltipMessage);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isLoading);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.icon != null);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.label, " ");
    } }, directives: [ɵngcc1.NgIf], styles: [".glyphicon-refresh-animate[_ngcontent-%COMP%] {\n    -animation: spin .7s infinite linear;\n    -webkit-animation: spin2 .7s infinite linear;\n}\n\n@-webkit-keyframes spin2 {\n    from { -webkit-transform: rotate(0deg);}\n    to { -webkit-transform: rotate(360deg);}\n}\n\n@keyframes spin {\n    from { transform: scale(1) rotate(0deg);}\n    to { transform: scale(1) rotate(360deg);}\n}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ButtonMediaComponent, [{
        type: _angular_core.Component,
        args: [{
                selector: 'amexio-ee-btn',
                template: "      \n          <button type=\"button\" (click)=\"btnClick($event)\"\n                  [class]=\"btnStyleClass\"\n                  [attr.fieldName] = \"fieldName\"\n                  [attr.disabled] = \"disabled ? true: null\"\n                  data-toggle=\"tooltip\" [attr.data-placement]=\"popoverPlacement\" [attr.title]=\"tooltipMessage\">\n              <ng-container *ngIf=\"isLoading\">\n                <i class=\"fa fa-refresh fa-spin \" aria-hidden=\"true\"></i>&nbsp;\n              </ng-container>\n              \n              <ng-container *ngIf=\"icon!=null\">\n                <i [class]=\"iconStyleClass\" aria-hidden=\"true\"></i>                \n              </ng-container>\n              {{label}}\n          </button>\n  ",
                styles: ["\n  .glyphicon-refresh-animate {\n    -animation: spin .7s infinite linear;\n    -webkit-animation: spin2 .7s infinite linear;\n}\n\n@-webkit-keyframes spin2 {\n    from { -webkit-transform: rotate(0deg);}\n    to { -webkit-transform: rotate(360deg);}\n}\n\n@keyframes spin {\n    from { transform: scale(1) rotate(0deg);}\n    to { transform: scale(1) rotate(360deg);}\n}"
                ]
            }]
    }], function () { return []; }, { onClick: [{
            type: _angular_core.Output
        }], popoverPlacement: [{
            type: _angular_core.Input
        }], label: [{
            type: _angular_core.Input
        }], icon: [{
            type: _angular_core.Input
        }], type: [{
            type: _angular_core.Input
        }], onClickRoute: [{
            type: _angular_core.Input
        }], tooltipMessage: [{
            type: _angular_core.Input
        }], disabled: [{
            type: _angular_core.Input
        }], isLoading: [{
            type: _angular_core.Input
        }], size: [{
            type: _angular_core.Input
        }], block: [{
            type: _angular_core.Input
        }], fieldName: [{
            type: _angular_core.Input
        }], cClass: [{
            type: _angular_core.Input
        }] }); })();
    return ButtonMediaComponent;
}());
/**
 * @nocollapse
 */
ButtonMediaComponent.ctorParameters = function () { return []; };
ButtonMediaComponent.propDecorators = {
    'label': [{ type: _angular_core.Input },],
    'icon': [{ type: _angular_core.Input },],
    'type': [{ type: _angular_core.Input },],
    'onClickRoute': [{ type: _angular_core.Input },],
    'tooltipMessage': [{ type: _angular_core.Input },],
    'disabled': [{ type: _angular_core.Input },],
    'isLoading': [{ type: _angular_core.Input },],
    'size': [{ type: _angular_core.Input },],
    'block': [{ type: _angular_core.Input },],
    'fieldName': [{ type: _angular_core.Input },],
    'popoverPlacement': [{ type: _angular_core.Input },],
    'cClass': [{ type: _angular_core.Input },],
    'onClick': [{ type: _angular_core.Output },],
};

/*
 * Copyright 2016-2017 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Author -  Pratik Kelwalkar
 *
 */
var RatingMediaComponent = (function () {
    function RatingMediaComponent() {
        // -------------------------------------------------------------------------
        // Inputs
        // -------------------------------------------------------------------------
        this.iconClass = "amexio-star-icon";
        this.fullIcon = "★";
        this.emptyIcon = "☆";
        this.titles = [];
        // -------------------------------------------------------------------------
        // Outputs
        // -------------------------------------------------------------------------
        this.onHover = new _angular_core.EventEmitter();
        this.onLeave = new _angular_core.EventEmitter();
        this.hovered = 0;
        this.hoveredPercent = undefined;
        this._max = 5;
    }
    Object.defineProperty(RatingMediaComponent.prototype, "max", {
        /**
         * @return {?}
         */
        get: function () {
            return this._max;
        },
        /**
         * @param {?} max
         * @return {?}
         */
        set: function (max) {
            this._max = max;
            this.buildRanges();
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} value
     * @return {?}
     */
    RatingMediaComponent.prototype.writeValue = function (value) {
        /*if (value % 1 !== value) {
         this.model = Math.round(value);
         return;
         }*/
        this.model = value;
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    RatingMediaComponent.prototype.registerOnChange = function (fn) {
        this.onChange = fn;
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    RatingMediaComponent.prototype.registerOnTouched = function (fn) {
        this.onTouched = fn;
    };
    /**
     * @param {?} c
     * @return {?}
     */
    RatingMediaComponent.prototype.validate = function (c) {
        if (this.required && !c.value) {
            return {
                required: true
            };
        }
        return null;
    };
    /**
     * @return {?}
     */
    RatingMediaComponent.prototype.ngOnInit = function () {
        this.buildRanges();
    };
    /**
     * @param {?} event
     * @return {?}
     */
    RatingMediaComponent.prototype.onKeydown = function (event) {
        if ([37, 38, 39, 40].indexOf(event.which) === -1 || this.hovered)
            return;
        event.preventDefault();
        event.stopPropagation();
        var /** @type {?} */ increment = this.float ? 0.5 : 1;
        this.rate(this.model + (event.which === 38 || event.which === 39 ? increment : increment * -1));
    };
    /**
     * @param {?} item
     * @return {?}
     */
    RatingMediaComponent.prototype.calculateWidth = function (item) {
        if (this.hovered > 0) {
            if (this.hoveredPercent !== undefined && this.hovered === item)
                return this.hoveredPercent;
            return this.hovered >= item ? 100 : 0;
        }
        return this.model >= item ? 100 : 100 - Math.round((item - this.model) * 10) * 10;
    };
    /**
     * @param {?} hovered
     * @return {?}
     */
    RatingMediaComponent.prototype.setHovered = function (hovered) {
        if (!this.readonly && !this.disabled) {
            this.hovered = hovered;
            this.onHover.emit(hovered);
        }
    };
    /**
     * @param {?} event
     * @return {?}
     */
    RatingMediaComponent.prototype.changeHovered = function (event) {
        if (!this.float)
            return;
        var /** @type {?} */ target = (event.target);
        var /** @type {?} */ relativeX = event.pageX - target.offsetLeft;
        var /** @type {?} */ percent = Math.round((relativeX * 100 / target.offsetWidth) / 10) * 10;
        this.hoveredPercent = percent > 50 ? 100 : 50;
    };
    /**
     * @return {?}
     */
    RatingMediaComponent.prototype.resetHovered = function () {
        this.hovered = 0;
        this.hoveredPercent = undefined;
        this.onLeave.emit(this.hovered);
    };
    /**
     * @param {?} value
     * @return {?}
     */
    RatingMediaComponent.prototype.rate = function (value) {
        if (!this.readonly && !this.disabled && value >= 0 && value <= this.ratingRange.length) {
            var /** @type {?} */ newValue = this.hoveredPercent ? (value - 1) + this.hoveredPercent / 100 : value;
            this.onChange(newValue);
            this.model = newValue;
        }
    };
    /**
     * @return {?}
     */
    RatingMediaComponent.prototype.buildRanges = function () {
        this.ratingRange = this.range(1, this.max);
    };
    /**
     * @param {?} start
     * @param {?} end
     * @return {?}
     */
    RatingMediaComponent.prototype.range = function (start, end) {
        var /** @type {?} */ foo = [];
        for (var /** @type {?} */ i = start; i <= end; i++) {
            foo.push(i);
        }
        return foo;
    };
RatingMediaComponent.ɵfac = function RatingMediaComponent_Factory(t) { return new (t || RatingMediaComponent)(); };
RatingMediaComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: RatingMediaComponent, selectors: [["amexio-ee-rating-input"]], hostBindings: function RatingMediaComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("keydown", function RatingMediaComponent_keydown_HostBindingHandler($event) { return ctx.onKeydown($event); });
    } }, inputs: { iconClass: "iconClass", fullIcon: "fullIcon", emptyIcon: "emptyIcon", titles: "titles", max: "max", readonly: "readonly", disabled: "disabled", required: "required", float: "float" }, outputs: { onHover: "onHover", onLeave: "onLeave" }, features: [ɵngcc0.ɵɵProvidersFeature([
            { provide: _angular_forms.NG_VALUE_ACCESSOR, useExisting: _angular_core.forwardRef(function () { return RatingMediaComponent; }), multi: true },
            { provide: _angular_forms.NG_VALIDATORS, useExisting: _angular_core.forwardRef(function () { return RatingMediaComponent; }), multi: true },
        ])], decls: 2, vars: 7, consts: [["tabindex", "0", "role", "slider", "aria-valuemin", "0", 1, "amexio-rating", 3, "mouseleave"], [4, "ngFor", "ngForOf"], [3, "title", "mouseenter", "mousemove", "click"]], template: function RatingMediaComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "span", 0);
        ɵngcc0.ɵɵlistener("mouseleave", function RatingMediaComponent_Template_span_mouseleave_0_listener() { return ctx.resetHovered(); });
        ɵngcc0.ɵɵtemplate(1, RatingMediaComponent_span_1_Template, 3, 7, "span", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassProp("disabled", ctx.disabled)("readonly", ctx.readonly);
        ɵngcc0.ɵɵattribute("aria-valuemax", ctx.ratingRange.length)("aria-valuenow", ctx.model);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.ratingRange);
    } }, directives: [ɵngcc1.NgForOf], styles: [".has-feedback-custom[_ngcontent-%COMP%] {\n          position: relative;\n      }\n      .has-feedback-custom[_ngcontent-%COMP%]   .form-control[_ngcontent-%COMP%] {\n          padding-right: 47.5px;\n      }\n\n      .form-control-feedback-custom[_ngcontent-%COMP%] {\n          position: absolute;\n          top: 0;\n          right: 0;\n          z-index: 2;\n          display: block;\n          width: 38px;\n          height: 38px;\n          line-height: 38px;\n          text-align: center;\n          pointer-events: none;\n      }\n\n      .has-feedback-custom[_ngcontent-%COMP%]   label[_ngcontent-%COMP%]    ~ .form-control-feedback-custom[_ngcontent-%COMP%] {\n          top: 32px;\n      }\n      .has-feedback-custom[_ngcontent-%COMP%]   label.sr-only[_ngcontent-%COMP%]    ~ .form-control-feedback-custom[_ngcontent-%COMP%] {\n          top: 0;\n      }\n      .amexio-rating[_ngcontent-%COMP%] {\n          cursor: pointer;\n          outline: none;\n      }\n      .amexio-rating.readonly[_ngcontent-%COMP%] {\n          cursor: default;\n      }\n      .amexio-rating.disabled[_ngcontent-%COMP%] {\n          cursor: not-allowed;\n      }\n      .amexio-rating[_ngcontent-%COMP%]   i[_ngcontent-%COMP%]{\n          font-style: normal;\n      }\n      .amexio-star-icon[_ngcontent-%COMP%] {\n          color: #666666;\n          font-size: 2em;\n          position: relative;\n          border-color: 1px solid black;\n      }\n      .amexio-star-icon[_ngcontent-%COMP%]:before {\n          color: #FDE16D;\n          content: attr(data-icon) \" \";\n          position: absolute;\n          left: 0;\n          overflow: hidden;\n          width: 0;\n      }\n      .amexio-rating.disabled[_ngcontent-%COMP%]   .amexio-star-icon[_ngcontent-%COMP%]:before {\n          color: #ECECEC;\n          text-shadow: none;\n      }\n      .amexio-star-icon.half10[_ngcontent-%COMP%]:before {\n          width: 10%;\n      }\n      .amexio-star-icon.half20[_ngcontent-%COMP%]:before {\n          width: 20%;\n      }\n      .amexio-star-icon.half30[_ngcontent-%COMP%]:before {\n          width: 30%;\n      }\n      .amexio-star-icon.half40[_ngcontent-%COMP%]:before {\n          width: 40%;\n      }\n      .amexio-star-icon.half50[_ngcontent-%COMP%]:before {\n          width: 50%;\n      }\n      .amexio-star-icon.half60[_ngcontent-%COMP%]:before {\n          width: 60%;\n      }\n      .amexio-star-icon.half70[_ngcontent-%COMP%]:before {\n          width: 70%;\n      }\n      .amexio-star-icon.half80[_ngcontent-%COMP%]:before {\n          width: 80%;\n      }\n      .amexio-star-icon.half90[_ngcontent-%COMP%]:before {\n          width: 90%;\n      }\n      .amexio-star-icon.half100[_ngcontent-%COMP%]:before {\n          width: 100%;\n      }\n      @-moz-document url-prefix() { \n          .amexio-star-icon {\n              font-size: 50px;\n              line-height: 34px;\n          }\n      }"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RatingMediaComponent, [{
        type: _angular_core.Component,
        args: [{
                selector: 'amexio-ee-rating-input',
                template: "\n    <span (mouseleave)=\"resetHovered()\"\n          class=\"amexio-rating\"\n          [class.disabled]=\"disabled\"\n          [class.readonly]=\"readonly\"\n          tabindex=\"0\"\n          role=\"slider\"\n          aria-valuemin=\"0\"\n          [attr.aria-valuemax]=\"ratingRange.length\"\n          [attr.aria-valuenow]=\"model\">\n    <span *ngFor=\"let item of ratingRange; let index = index\">\n        <i (mouseenter)=\"setHovered(item)\"\n           (mousemove)=\"changeHovered($event)\"\n           (click)=\"rate(item)\"\n           [attr.data-icon]=\"fullIcon\"\n           class=\"{{ iconClass }} half{{ calculateWidth(item) }}\"\n           [title]=\"titles[index] || item\">{{ emptyIcon }}</i>\n    </span>\n</span>\n  ",
                providers: [
                    { provide: _angular_forms.NG_VALUE_ACCESSOR, useExisting: _angular_core.forwardRef(function () { return RatingMediaComponent; }), multi: true },
                    { provide: _angular_forms.NG_VALIDATORS, useExisting: _angular_core.forwardRef(function () { return RatingMediaComponent; }), multi: true },
                ],
                styles: [
                    "/**\n A Style Sheet for all form inputs common used classes\n */\n\n      /** Form Validations & Icon Positioning **/\n      .has-feedback-custom {\n          position: relative;\n      }\n      .has-feedback-custom .form-control {\n          padding-right: 47.5px;\n      }\n\n      .form-control-feedback-custom {\n          position: absolute;\n          top: 0;\n          right: 0;\n          z-index: 2;\n          display: block;\n          width: 38px;\n          height: 38px;\n          line-height: 38px;\n          text-align: center;\n          pointer-events: none;\n      }\n\n      .has-feedback-custom label ~ .form-control-feedback-custom {\n          top: 32px;\n      }\n      .has-feedback-custom label.sr-only ~ .form-control-feedback-custom {\n          top: 0;\n      }\n      .amexio-rating {\n          cursor: pointer;\n          outline: none;\n      }\n      .amexio-rating.readonly {\n          cursor: default;\n      }\n      .amexio-rating.disabled {\n          cursor: not-allowed;\n      }\n      .amexio-rating i{\n          font-style: normal;\n      }\n      .amexio-star-icon {\n          color: #666666;\n          font-size: 2em;\n          position: relative;\n          border-color: 1px solid black;\n      }\n      .amexio-star-icon:before {\n          color: #FDE16D;\n          content: attr(data-icon) \" \";\n          position: absolute;\n          left: 0;\n          overflow: hidden;\n          width: 0;\n      }\n      .amexio-rating.disabled .amexio-star-icon:before {\n          color: #ECECEC;\n          text-shadow: none;\n      }\n      .amexio-star-icon.half10:before {\n          width: 10%;\n      }\n      .amexio-star-icon.half20:before {\n          width: 20%;\n      }\n      .amexio-star-icon.half30:before {\n          width: 30%;\n      }\n      .amexio-star-icon.half40:before {\n          width: 40%;\n      }\n      .amexio-star-icon.half50:before {\n          width: 50%;\n      }\n      .amexio-star-icon.half60:before {\n          width: 60%;\n      }\n      .amexio-star-icon.half70:before {\n          width: 70%;\n      }\n      .amexio-star-icon.half80:before {\n          width: 80%;\n      }\n      .amexio-star-icon.half90:before {\n          width: 90%;\n      }\n      .amexio-star-icon.half100:before {\n          width: 100%;\n      }\n      @-moz-document url-prefix() { /* Firefox Hack */\n          .amexio-star-icon {\n              font-size: 50px;\n              line-height: 34px;\n          }\n      }\n    "
                ]
            }]
    }], function () { return []; }, { iconClass: [{
            type: _angular_core.Input
        }], fullIcon: [{
            type: _angular_core.Input
        }], emptyIcon: [{
            type: _angular_core.Input
        }], titles: [{
            type: _angular_core.Input
        }], onHover: [{
            type: _angular_core.Output
        }], onLeave: [{
            type: _angular_core.Output
        }], max: [{
            type: _angular_core.Input
        }], onKeydown: [{
            type: _angular_core.HostListener,
            args: ["keydown", ["$event"]]
        }], readonly: [{
            type: _angular_core.Input
        }], disabled: [{
            type: _angular_core.Input
        }], required: [{
            type: _angular_core.Input
        }], float: [{
            type: _angular_core.Input
        }] }); })();
    return RatingMediaComponent;
}());
/**
 * @nocollapse
 */
RatingMediaComponent.ctorParameters = function () { return []; };
RatingMediaComponent.propDecorators = {
    'iconClass': [{ type: _angular_core.Input },],
    'fullIcon': [{ type: _angular_core.Input },],
    'emptyIcon': [{ type: _angular_core.Input },],
    'readonly': [{ type: _angular_core.Input },],
    'disabled': [{ type: _angular_core.Input },],
    'required': [{ type: _angular_core.Input },],
    'float': [{ type: _angular_core.Input },],
    'titles': [{ type: _angular_core.Input },],
    'max': [{ type: _angular_core.Input },],
    'onHover': [{ type: _angular_core.Output },],
    'onLeave': [{ type: _angular_core.Output },],
    'onKeydown': [{ type: _angular_core.HostListener, args: ["keydown", ["$event"],] },],
};

/*
 * Copyright 2016-2017 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Author - Pratik Kelwalkar
 *
 */
var AmexioYoutubePlayerComponent = (function () {
    function AmexioYoutubePlayerComponent() {
        this.onCloseVideoPlayer = new _angular_core.EventEmitter();
    }
    /**
     * @return {?}
     */
    AmexioYoutubePlayerComponent.prototype.ngOnInit = function () {
        if (this.height == null || this.height === 'undefined') {
            this.height = '98%';
        }
    };
    /**
     * @return {?}
     */
    AmexioYoutubePlayerComponent.prototype.routeBackToApp = function () {
        this.onCloseVideoPlayer.emit(this.url);
    };
AmexioYoutubePlayerComponent.ɵfac = function AmexioYoutubePlayerComponent_Factory(t) { return new (t || AmexioYoutubePlayerComponent)(); };
AmexioYoutubePlayerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AmexioYoutubePlayerComponent, selectors: [["amexio-ee-youtube-player"]], inputs: { height: "height", url: "url" }, outputs: { onCloseVideoPlayer: "onCloseVideoPlayer" }, decls: 5, vars: 2, consts: [[1, "close-button"], ["aria-hidden", "true", 1, "fa", "fa-times", "fa-lg", 3, "click"], [1, "text-center"], ["width", "98%", "frameborder", "0", "allowfullscreen", "", 2, "position", "absolute", "top", "0", "right", "0", "bottom", "0", "left", "0", "padding-top", "30px", "padding-left", "30px", 3, "height", "src"]], template: function AmexioYoutubePlayerComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵelementStart(1, "span", 0);
        ɵngcc0.ɵɵelementStart(2, "i", 1);
        ɵngcc0.ɵɵlistener("click", function AmexioYoutubePlayerComponent_Template_i_click_2_listener() { return ctx.routeBackToApp(); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 2);
        ɵngcc0.ɵɵelement(4, "iframe", 3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵproperty("height", ctx.height)("src", ctx.url, ɵngcc0.ɵɵsanitizeResourceUrl);
    } }, styles: [".close-button[_ngcontent-%COMP%]{\n            cursor: pointer;\n            padding-left: 98%;color:gray;background: radial-gradient(ellipse at top right,rgba(0,0,0,.4) 0,rgba(0,0,0,0)70%,rgba(0,0,0,0) 100%);\n        }\n        .close-button[_ngcontent-%COMP%]   i[_ngcontent-%COMP%]:hover{\n            color: red;\n            cursor: pointer;\n        }"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AmexioYoutubePlayerComponent, [{
        type: _angular_core.Component,
        args: [{
                selector: 'amexio-ee-youtube-player',
                template: "\n        <div >\n      <span  class=\"close-button\">\n        <i class=\"fa fa-times fa-lg\" (click)=\"routeBackToApp()\" aria-hidden=\"true\"></i>\n      </span>\n            <div class=\"text-center\">\n                <iframe  [height]=\"height\" width=\"98%\"\n                         style=\"position: absolute; top: 0; right: 0; bottom: 0; left: 0;padding-top: 30px;padding-left: 30px\"\n                         [src]=\"url\" frameborder=\"0\"\n                         allowfullscreen>\n                </iframe>\n            </div>\n        </div>\n    ",
                styles: ["\n        .close-button{\n            cursor: pointer;\n            padding-left: 98%;color:gray;background: radial-gradient(ellipse at top right,rgba(0,0,0,.4) 0,rgba(0,0,0,0)70%,rgba(0,0,0,0) 100%);\n        }\n        .close-button i:hover{\n            color: red;\n            cursor: pointer;\n        }\n    "]
            }]
    }], function () { return []; }, { onCloseVideoPlayer: [{
            type: _angular_core.Output
        }], height: [{
            type: _angular_core.Input
        }], url: [{
            type: _angular_core.Input
        }] }); })();
    return AmexioYoutubePlayerComponent;
}());
/**
 * @nocollapse
 */
AmexioYoutubePlayerComponent.ctorParameters = function () { return []; };
AmexioYoutubePlayerComponent.propDecorators = {
    'url': [{ type: _angular_core.Input },],
    'height': [{ type: _angular_core.Input },],
    'onCloseVideoPlayer': [{ type: _angular_core.Output },],
};

/*
 * Copyright 2016-2017 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Author - Pratik Kelwalkar
 *
 */
var MultiMediaCarousel = (function () {
    function MultiMediaCarousel() {
        this.hasDetail = true;
        this.onVideoLoad = new _angular_core.EventEmitter();
        this.right = false;
        this.left = false;
        this.elementId = 'multi-media-ee-carousel-' + Math.floor(Math.random() * 90000) + 10000;
    }
    /**
     * @return {?}
     */
    MultiMediaCarousel.prototype.ngOnInit = function () {
        if (this.carouselStyle == null) {
            this.carouselStyle = 'horizontal';
        }
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    MultiMediaCarousel.prototype.ngOnChanges = function (changes) {
        if (changes['data'] != null) {
            this.data = changes['data'].currentValue;
        }
    };
    /**
     * @return {?}
     */
    MultiMediaCarousel.prototype.leftClick = function () {
        var /** @type {?} */ ts = document.getElementById(this.elementId);
        $('#' + this.elementId).animate({ scrollLeft: '-=350' }, 600);
        if (ts.scrollLeft == 0) {
            document.getElementById(this.elementId + 'leftarrow').style.visibility = 'hidden';
        }
        document.getElementById(this.elementId + 'rightarrow').style.visibility = 'visible';
    };
    /**
     * @return {?}
     */
    MultiMediaCarousel.prototype.rightClick = function () {
        var /** @type {?} */ ts = document.getElementById(this.elementId);
        $('#' + this.elementId).animate({ scrollLeft: '+=350' }, 600);
        if ((ts.scrollWidth - ts.offsetWidth - ts.scrollLeft) <= 0) {
            document.getElementById(this.elementId + 'rightarrow').style.visibility = 'hidden';
        }
        document.getElementById(this.elementId + 'leftarrow').style.visibility = 'visible';
    };
    /**
     * @return {?}
     */
    MultiMediaCarousel.prototype.closeDetailPage = function () {
        this.data.forEach(function (item) {
            item.selected = null;
        });
        $('#' + this.elementId + 'carousel-detail-content').collapse('hide');
    };
    /**
     * @param {?} item
     * @return {?}
     */
    MultiMediaCarousel.prototype.openDetailsSection = function (item) {
        var _this = this;
        this.videoUrl = item.video;
        this.currentDetailsImagePath = item.details_img;
        this.currentDetailsTitle = item.title;
        this.currentDetailsDesc = item.desc;
        this.currentContent = item.content;
        this.seasonNo = item.seasonNo;
        this.matchPercentage = item.matchPercentage;
        this.ageLimit = item.ageLimit;
        this.releaseYear = item.releaseYear;
        this.data.forEach(function (item) {
            item.selected = null;
        });
        item.selected = "selected";
        $('#' + this.elementId + 'carousel-detail-content').collapse('show');
        setTimeout(function () {
            $('html, body').animate({
                scrollTop: $('#' + _this.elementId + 'carousel-detail-content').offset().top
            }, 500);
        }, 500);
    };
    /**
     * @param {?} item
     * @return {?}
     */
    MultiMediaCarousel.prototype.loadVideo = function (item) {
        this.onVideoLoad.emit(item.video);
    };
    /**
     * @param {?} video
     * @return {?}
     */
    MultiMediaCarousel.prototype.playVideo = function (video) {
        this.onVideoLoad.emit(video);
    };
MultiMediaCarousel.ɵfac = function MultiMediaCarousel_Factory(t) { return new (t || MultiMediaCarousel)(); };
MultiMediaCarousel.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MultiMediaCarousel, selectors: [["amexio-media-ee-carousel"]], inputs: { hasDetail: "hasDetail", carouselStyle: "carouselStyle", data: "data", title: "title" }, outputs: { onVideoLoad: "onVideoLoad" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 13, vars: 14, consts: [[1, "section-header", 2, "z-index", "1"], [1, "contain", "row"], [1, "left-arrow", 2, "cursor", "pointer"], [3, "ngClass", "click"], ["aria-hidden", "true", 1, "fa", "fa-chevron-left", "fa-2x"], [1, "row1"], [1, "row__inner"], [4, "ngIf"], [1, "right-arrow", 2, "cursor", "pointer"], ["aria-hidden", "true", 1, "fa", "fa-chevron-right", "fa-2x"], ["class", "collapse", 4, "ngIf"], ["class", "tile", 4, "ngFor", "ngForOf"], [1, "tile"], [1, "tile__media", 3, "click"], [1, "tile__img"], ["aria-hidden", "false", 1, "fa", "fa-caret-down", 3, "ngStyle"], [3, "ngClass"], [1, "textonly"], [1, "playpanel"], [1, "playbutton", 2, "cursor", "pointer", 3, "click"], ["aria-hidden", "true", 1, "fa", "fa-play"], ["class", "title", 4, "ngIf"], [1, "match"], [1, "age", "mini"], [1, "subtitle"], ["class", "actionpanel", 4, "ngIf"], [1, "title"], [1, "actionpanel"], [2, "cursor", "pointer", 3, "click"], ["aria-hidden", "true", 1, "fa", "fa-chevron-down", "fa-2x"], [1, "collapse"], [3, "bgImgUrl", "title", "description", "contents", "videoLink", "rate", "enableWatch", "enableMyList", "closeEnable", "seasonNo", "releaseYear", "ageLimit", "matchPercentage", "onWatchClick", "onCloseClick"]], template: function MultiMediaCarousel_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtext(1);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(2, "div", 1);
        ɵngcc0.ɵɵelementStart(3, "div", 2);
        ɵngcc0.ɵɵelementStart(4, "div", 3);
        ɵngcc0.ɵɵlistener("click", function MultiMediaCarousel_Template_div_click_4_listener() { return ctx.leftClick(); });
        ɵngcc0.ɵɵelement(5, "i", 4);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(6, "div", 5);
        ɵngcc0.ɵɵelementStart(7, "div", 6);
        ɵngcc0.ɵɵtemplate(8, MultiMediaCarousel_ng_container_8_Template, 2, 1, "ng-container", 7);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(9, "div", 8);
        ɵngcc0.ɵɵelementStart(10, "div", 3);
        ɵngcc0.ɵɵlistener("click", function MultiMediaCarousel_Template_div_click_10_listener() { return ctx.rightClick(); });
        ɵngcc0.ɵɵelement(11, "i", 9);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(12, MultiMediaCarousel_div_12_Template, 2, 14, "div", 10);
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate(ctx.title);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵattribute("id", ctx.elementId + "leftarrow");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction2(8, _c3, ctx.carouselStyle == "horizontal", ctx.carouselStyle == "vertical"));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵattribute("id", ctx.elementId);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.data);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵattribute("id", ctx.elementId + "rightarrow");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction2(11, _c3, ctx.carouselStyle == "horizontal", ctx.carouselStyle == "vertical"));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.hasDetail);
    } }, directives: [ɵngcc1.NgClass, ɵngcc1.NgIf, ɵngcc1.NgForOf, ɵngcc1.NgStyle, ContentComponent], styles: ["*[_ngcontent-%COMP%] { box-sizing: border-box; } h1[_ngcontent-%COMP%], p[_ngcontent-%COMP%] { text-align: center; } p[_ngcontent-%COMP%] { width: 100%; max-width: 500px; margin: auto; } a[_ngcontent-%COMP%]:link, a[_ngcontent-%COMP%]:hover, a[_ngcontent-%COMP%]:active, a[_ngcontent-%COMP%]:visited { -webkit-transition: color 150ms; transition: color 150ms; color: #95a5a6; text-decoration: none; } a[_ngcontent-%COMP%]:hover { color: #7f8c8d; text-decoration: underline; } .contain[_ngcontent-%COMP%] { width: 100%; } .contain-vertical[_ngcontent-%COMP%]{ width: 100%; height: 450px; } .row1[_ngcontent-%COMP%] { overflow: hidden; width: 94%; } .row__inner[_ngcontent-%COMP%] { -webkit-transition: 450ms -webkit-transform; transition: 450ms -webkit-transform; transition: 450ms transform; transition: 450ms transform, 450ms -webkit-transform; font-size: 0; white-space: nowrap; margin: 70.3125px 0; padding-bottom: 10px; } .tile[_ngcontent-%COMP%] { position: relative; display: inline-block; width: 250px;  margin-right: 10px; font-size: 20px; cursor: pointer; -webkit-transition: 450ms all; transition: 450ms all; -webkit-transform-origin: center left; transform-origin: center left; } .tile__img[_ngcontent-%COMP%] { width: 250px;  -o-object-fit: cover; object-fit: cover; } .tile__details[_ngcontent-%COMP%] { position: absolute; bottom: 0; left: 0; right: 0; top: 0; font-size: 10px; opacity: 0; background: -webkit-linear-gradient(bottom, rgba(0,0,0,0.9) 0%, rgba(0,0,0,0) 100%); background: linear-gradient(to top, rgba(0,0,0,0.9) 0%, rgba(0,0,0,0) 100%); -webkit-transition: 450ms opacity; transition: 450ms opacity; } .tile__details[_ngcontent-%COMP%]:after, .tile__details[_ngcontent-%COMP%]:before { content: ''; position: absolute; top: 50%; left: 50%; display: #000; } .tile__details[_ngcontent-%COMP%]:after { margin-top: -25px; margin-left: -25px; width: 50px; height: 50px;  line-height: 50px; text-align: center; border-radius: 100%;  z-index: 1; } .tile__details[_ngcontent-%COMP%]:before {  } .tile__details_v[_ngcontent-%COMP%] { position: absolute; bottom: 0; left: 0; right: 0; top: 50%; font-size: 10px; opacity: 0; background: -webkit-linear-gradient(bottom, rgba(0,0,0,0.9) 0%, rgba(0,0,0,0) 100%); background: linear-gradient(to top, rgba(0,0,0,0.9) 0%, rgba(0,0,0,0) 100%); -webkit-transition: 450ms opacity; transition: 450ms opacity; } .tile__details_v[_ngcontent-%COMP%]:after, .tile__details_v[_ngcontent-%COMP%]:before { content: ''; position: absolute; top: 50%; left: 50%; display: #000; } .tile__details_v[_ngcontent-%COMP%]:after { margin-top: -25px; margin-left: -20px; width: 50px; height: 50px;  line-height: 50px; text-align: center; border-radius: 100%;  z-index: 1; } .tile__details_v[_ngcontent-%COMP%]:before {  } .tile[_ngcontent-%COMP%]:hover   .tile__details_v[_ngcontent-%COMP%] { opacity: 1; } .tile[_ngcontent-%COMP%]:hover   .tile__details[_ngcontent-%COMP%] { opacity: 1; } .tile__title[_ngcontent-%COMP%] { position: absolute; bottom: 0; padding: 10px; color: white; } .row__inner[_ngcontent-%COMP%]:hover { -webkit-transform: translate3d(-62.5px, 0, 0); transform: translate3d(-62.5px, 0, 0); } .row__inner[_ngcontent-%COMP%]:hover   .tile[_ngcontent-%COMP%] { opacity: 0.3; } .row__inner[_ngcontent-%COMP%]:hover   .tile[_ngcontent-%COMP%]:hover { -webkit-transform: scale(1.5); transform: scale(1.5); opacity: 1; } .tile[_ngcontent-%COMP%]:hover    ~ .tile[_ngcontent-%COMP%] { -webkit-transform: translate3d(125px, 0, 0); transform: translate3d(125px, 0, 0); } .left-arrow[_ngcontent-%COMP%]{ float: left; width: 4%; } .right-arrow[_ngcontent-%COMP%]{ width: 2%; } .controls[_ngcontent-%COMP%]{ padding-top: 125px; display: flex; align-items: center; justify-content: center; color: dimgrey; font-size: 1em; } .controls-v[_ngcontent-%COMP%]{ padding-top: 250px; display: flex; align-items: center; justify-content: center; color: dimgrey; font-size: 1em; } .textonly[_ngcontent-%COMP%]{ padding: 0px 7px 2px 7px; width: 100%; text-align: left; color: white; background-color: transparent; } .title[_ngcontent-%COMP%]{ font-size: 10pt; font-weight: 600; } .subtitle[_ngcontent-%COMP%]{ font-size: 6pt; font-weight: 500; white-space: pre-wrap; } .match[_ngcontent-%COMP%]{ color: lightgreen; } .age[_ngcontent-%COMP%]{ display: inline-block; border: solid 2px #666; padding: 0 10px 0 10px; font-size: 100%; color: white; } .mini[_ngcontent-%COMP%]{ border: solid 1px #666; padding: 0 5px 0 5px; } .actionpanel[_ngcontent-%COMP%]{ padding-top: 4px; padding-left: 109px; color:  #e50914; background-color: transparent; } .section-header[_ngcontent-%COMP%]{ font-weight: 700; padding: 0em 0em 0em 1em; font-size: 150%; color: white; position: absolute; bottom: 100; } .playpanel[_ngcontent-%COMP%]{ padding: 40px 0 0 100px; background-size: cover; background-position: center; } .playpanel[_ngcontent-%COMP%]   .playbutton[_ngcontent-%COMP%]{ display: inline-block; border: solid 1px #ffffff; padding: 8px  0px 0px 10px; height:30px; width: 30px; border-radius: 40px; background-color: rgba(0,0,0,.5); color: #fff; } .playpanel[_ngcontent-%COMP%]   .playbutton[_ngcontent-%COMP%]:hover{ color: #e50914; } .selected[_ngcontent-%COMP%]{ align-items: center; display: flex;justify-content: center; color: white; padding: 0px; height: 5px; border-top: 3px solid white; }"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MultiMediaCarousel, [{
        type: _angular_core.Component,
        args: [{
                selector: 'amexio-media-ee-carousel',
                template: "<div class=\"section-header\" style=\"z-index: 1;\">{{title}}</div> <div class=\"contain row\"> <div class=\"left-arrow\" [attr.id]=\"elementId+'leftarrow'\" style=\"cursor: pointer\"> <div [ngClass]=\"{'controls' : carouselStyle=='horizontal','controls-v' : carouselStyle=='vertical'}\" (click)=\"leftClick();\"><i class=\"fa fa-chevron-left fa-2x\" aria-hidden=\"true\"></i></div> </div> <div class=\"row1\" [attr.id]=\"elementId\"> <div class=\"row__inner\"> <ng-container *ngIf=\"data\"> <div class=\"tile\" *ngFor=\"let item of data;let i = index\"> <div class=\"tile__media\" (click)=\"loadVideo(item)\"> <img class=\"tile__img\" [attr.src]=\"item.img\" /> </div> <div class=\"{{item.selected}}\"> <i [ngStyle]=\"{'visibility':item.selected ? 'visible':'hidden'}\" class=\"fa fa-caret-down\" aria-hidden=\"false\"></i> </div> <div [ngClass]=\"{'tile__details' : carouselStyle=='horizontal','tile__details_v' : carouselStyle=='vertical'}\"> <div class=\"textonly\"> <div class=\"playpanel\"> <div style=\"cursor: pointer\" (click)=\"loadVideo(item)\" class=\"playbutton\"><i class=\"fa fa-play\" aria-hidden=\"true\"></i></div> </div> <span class=\"title\" *ngIf=\"item.title\">{{item.title}}</span><br> <span class=\"match\">{{item.matchPercentage}} Match</span> {{item.releaseYear}} <span class=\"age mini\">{{item.ageLimit}}</span> {{item.seasonNo}}<br> <div class=\"subtitle\">{{item.desc}}</div> </div> <div class=\"actionpanel\"   *ngIf=\"item.content\"> <div style=\"cursor: pointer;\" (click)=\"openDetailsSection(item)\"><i class=\"fa fa-chevron-down fa-2x\" aria-hidden=\"true\"></i></div> </div> </div> </div> </ng-container> </div> </div> <div class=\"right-arrow\" [attr.id]=\"elementId+'rightarrow'\" style=\"cursor: pointer\"> <div [ngClass]=\"{'controls' : carouselStyle=='horizontal','controls-v' : carouselStyle=='vertical'}\" (click)=\"rightClick();\"><i class=\"fa fa-chevron-right fa-2x\" aria-hidden=\"true\"></i></div></div> </div> <div [attr.id]=\"elementId+'carousel-detail-content'\" class=\"collapse\" *ngIf=\"hasDetail\"> <amexio-ee-content [bgImgUrl]=\"currentDetailsImagePath\" [title]=\"currentDetailsTitle\" [description]=\"currentDetailsDesc\" [contents]=\"currentContent\" [videoLink]=\"videoUrl\" (onWatchClick)=\"playVideo(videoUrl)\" (onCloseClick)=\"closeDetailPage()\" [rate]=\"rate\" [enableWatch]=\"true\" [enableMyList]=\"true\" [closeEnable]=\"true\" [seasonNo]=\"seasonNo\" [releaseYear]=\"releaseYear\" [ageLimit]=\"ageLimit\" [matchPercentage]=\"matchPercentage\" ></amexio-ee-content> </div> ",
                styles: [" * { box-sizing: border-box; } h1, p { text-align: center; } p { width: 100%; max-width: 500px; margin: auto; } a:link, a:hover, a:active, a:visited { -webkit-transition: color 150ms; transition: color 150ms; color: #95a5a6; text-decoration: none; } a:hover { color: #7f8c8d; text-decoration: underline; } .contain { width: 100%; } .contain-vertical{ width: 100%; height: 450px; } .row1 { overflow: hidden; width: 94%; } .row__inner { -webkit-transition: 450ms -webkit-transform; transition: 450ms -webkit-transform; transition: 450ms transform; transition: 450ms transform, 450ms -webkit-transform; font-size: 0; white-space: nowrap; margin: 70.3125px 0; padding-bottom: 10px; } .tile { position: relative; display: inline-block; width: 250px; /*height: 140.625px;*/ margin-right: 10px; font-size: 20px; cursor: pointer; -webkit-transition: 450ms all; transition: 450ms all; -webkit-transform-origin: center left; transform-origin: center left; } .tile__img { width: 250px; /* height: 140.625px;*/ -o-object-fit: cover; object-fit: cover; } .tile__details { position: absolute; bottom: 0; left: 0; right: 0; top: 0; font-size: 10px; opacity: 0; background: -webkit-linear-gradient(bottom, rgba(0,0,0,0.9) 0%, rgba(0,0,0,0) 100%); background: linear-gradient(to top, rgba(0,0,0,0.9) 0%, rgba(0,0,0,0) 100%); -webkit-transition: 450ms opacity; transition: 450ms opacity; } .tile__details:after, .tile__details:before { content: ''; position: absolute; top: 50%; left: 50%; display: #000; } .tile__details:after { margin-top: -25px; margin-left: -25px; width: 50px; height: 50px; /*border: 3px solid #ecf0f1;*/ line-height: 50px; text-align: center; border-radius: 100%; /*background: rgba(0,0,0,0.5);*/ z-index: 1; } .tile__details:before { /* content: '▶'; left: 0; width: 100%; font-size: 30px; margin-left: 7px; margin-top: -24px; text-align: center; z-index: 2; color: red;*/ } .tile__details_v { position: absolute; bottom: 0; left: 0; right: 0; top: 50%; font-size: 10px; opacity: 0; background: -webkit-linear-gradient(bottom, rgba(0,0,0,0.9) 0%, rgba(0,0,0,0) 100%); background: linear-gradient(to top, rgba(0,0,0,0.9) 0%, rgba(0,0,0,0) 100%); -webkit-transition: 450ms opacity; transition: 450ms opacity; } .tile__details_v:after, .tile__details_v:before { content: ''; position: absolute; top: 50%; left: 50%; display: #000; } .tile__details_v:after { margin-top: -25px; margin-left: -20px; width: 50px; height: 50px; /*border: 3px solid #ecf0f1;*/ line-height: 50px; text-align: center; border-radius: 100%; /*background: rgba(0,0,0,0.5);*/ z-index: 1; } .tile__details_v:before { /* content: '▶'; left: 0; width: 100%; font-size: 30px; margin-left: 7px; margin-top: -24px; text-align: center; z-index: 2; color: red;*/ } .tile:hover .tile__details_v { opacity: 1; } .tile:hover .tile__details { opacity: 1; } .tile__title { position: absolute; bottom: 0; padding: 10px; color: white; } .row__inner:hover { -webkit-transform: translate3d(-62.5px, 0, 0); transform: translate3d(-62.5px, 0, 0); } .row__inner:hover .tile { opacity: 0.3; } .row__inner:hover .tile:hover { -webkit-transform: scale(1.5); transform: scale(1.5); opacity: 1; } .tile:hover ~ .tile { -webkit-transform: translate3d(125px, 0, 0); transform: translate3d(125px, 0, 0); } .left-arrow{ float: left; width: 4%; } .right-arrow{ width: 2%; } .controls{ padding-top: 125px; display: flex; align-items: center; justify-content: center; color: dimgrey; font-size: 1em; } .controls-v{ padding-top: 250px; display: flex; align-items: center; justify-content: center; color: dimgrey; font-size: 1em; } .textonly{ padding: 0px 7px 2px 7px; width: 100%; text-align: left; color: white; background-color: transparent; } .title{ font-size: 10pt; font-weight: 600; } .subtitle{ font-size: 6pt; font-weight: 500; white-space: pre-wrap; } .match{ color: lightgreen; } .age{ display: inline-block; border: solid 2px #666; padding: 0 10px 0 10px; font-size: 100%; color: white; } .mini{ border: solid 1px #666; padding: 0 5px 0 5px; } .actionpanel{ padding-top: 4px; padding-left: 109px; color:  #e50914; background-color: transparent; } .section-header{ font-weight: 700; padding: 0em 0em 0em 1em; font-size: 150%; color: white; position: absolute; bottom: 100; } .playpanel{ padding: 40px 0 0 100px; background-size: cover; background-position: center; } .playpanel .playbutton{ display: inline-block; border: solid 1px #ffffff; padding: 8px  0px 0px 10px; height:30px; width: 30px; border-radius: 40px; background-color: rgba(0,0,0,.5); color: #fff; } .playpanel .playbutton:hover{ color: #e50914; } .selected{ align-items: center; display: flex;justify-content: center; color: white; padding: 0px; height: 5px; border-top: 3px solid white; } "]
            }]
    }], function () { return []; }, { hasDetail: [{
            type: _angular_core.Input
        }], onVideoLoad: [{
            type: _angular_core.Output
        }], carouselStyle: [{
            type: _angular_core.Input
        }], data: [{
            type: _angular_core.Input
        }], title: [{
            type: _angular_core.Input
        }] }); })();
    return MultiMediaCarousel;
}());
/**
 * @nocollapse
 */
MultiMediaCarousel.ctorParameters = function () { return []; };
MultiMediaCarousel.propDecorators = {
    'data': [{ type: _angular_core.Input },],
    'carouselStyle': [{ type: _angular_core.Input },],
    'hasDetail': [{ type: _angular_core.Input },],
    'title': [{ type: _angular_core.Input },],
    'onVideoLoad': [{ type: _angular_core.Output },],
};

var AmexioMediaModule = (function () {
    function AmexioMediaModule() {
    }
    /**
     * @return {?}
     */
    AmexioMediaModule.forRoot = function () {
        return {
            ngModule: AmexioMediaModule,
            providers: []
        };
    };
AmexioMediaModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AmexioMediaModule });
AmexioMediaModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function AmexioMediaModule_Factory(t) { return new (t || AmexioMediaModule)(); }, imports: [[
            _angular_common.CommonModule,
            _angular_http.HttpModule,
            _angular_forms.FormsModule,
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AmexioMediaModule, { declarations: [ContentComponent, ButtonMediaComponent, RatingMediaComponent, AmexioYoutubePlayerComponent, MultiMediaCarousel], imports: [ɵngcc1.CommonModule, ɵngcc3.HttpModule, ɵngcc2.FormsModule], exports: [ContentComponent, ButtonMediaComponent, RatingMediaComponent, AmexioYoutubePlayerComponent, MultiMediaCarousel] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AmexioMediaModule, [{
        type: _angular_core.NgModule,
        args: [{
                imports: [
                    _angular_common.CommonModule,
                    _angular_http.HttpModule,
                    _angular_forms.FormsModule,
                ],
                declarations: [
                    ContentComponent,
                    ButtonMediaComponent,
                    RatingMediaComponent,
                    AmexioYoutubePlayerComponent,
                    MultiMediaCarousel
                ],
                exports: [
                    ContentComponent,
                    ButtonMediaComponent,
                    RatingMediaComponent,
                    AmexioYoutubePlayerComponent,
                    MultiMediaCarousel
                ]
            }]
    }], function () { return []; }, null); })();
    return AmexioMediaModule;
}());
/**
 * @nocollapse
 */
AmexioMediaModule.ctorParameters = function () { return []; };

exports.AmexioMediaModule = AmexioMediaModule;
exports.ContentComponent = ContentComponent;
exports.ButtonMediaComponent = ButtonMediaComponent;
exports.RatingMediaComponent = RatingMediaComponent;
exports.MultiMediaCarousel = MultiMediaCarousel;
exports.AmexioYoutubePlayerComponent = AmexioYoutubePlayerComponent;

Object.defineProperty(exports, '__esModule', { value: true });

})));
