// countup.js-angular2/esm5/lib/count-up.module.js
import * as tslib_1 from "tslib";
import { NgModule } from '@angular/core';
import { CountUpDirective } from './count-up.directive';
import * as ɵngcc0 from '@angular/core';
var CountUpModule = /** @class */ (function () {
    function CountUpModule() {
    }
CountUpModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CountUpModule });
CountUpModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function CountUpModule_Factory(t) { return new (t || CountUpModule)(); }, imports: [[]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CountUpModule, { declarations: function () { return [CountUpDirective]; }, exports: function () { return [CountUpDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CountUpModule, [{
        type: NgModule,
        args: [{
                imports: [],
                declarations: [CountUpDirective],
                exports: [CountUpDirective]
            }]
    }], function () { return []; }, null); })();
    return CountUpModule;
}());
export { CountUpModule };
